\documentclass[10pt,A4paper]{article}
\usepackage[utf8]{inputenc}

\setlength{\tabcolsep}{1.3pt}
\setlength{\columnsep}{0.4cm}
\usepackage{acronym}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage[
    backend=biber,
    sorting=none,
    style=numeric-comp,
    hyperref=true,
    maxnames=2,
    url=false,
    isbn=false,
    eprint=false,
    date=year,
    doi=false,
    giveninits=true
]{biblatex}
\usepackage{array}
\usepackage{bm}
\usepackage{booktabs}
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off, font=small]{caption}
\usepackage{color}
\usepackage{diagbox}
\usepackage{float}
\usepackage[T1]{fontenc}
\usepackage[a4paper,inner=2cm,outer=2cm,top=2.5cm,bottom=2.5cm]{geometry}
\usepackage{graphicx}% Include figure files
\usepackage{here}
\usepackage{hyperref}
\usepackage{lipsum}
\usepackage{listings}
\usepackage{listings-rust}
\usepackage{mathtools}
\usepackage{mathrsfs}
\usepackage{multirow}
\usepackage{newfloat}
\usepackage{onimage}
\usepackage{siunitx}
\usepackage{subcaption}
\usepackage{tabularx}
\usepackage{tikz}
\usepackage[dvipsnames]{xcolor}
\usepackage{xpatch}

% Must be loaded after tikz
\usepackage{ctable}

% Define new column type with multirow
\newcolumntype{M}[1]{>{\centering\arraybackslash}m{#1}}

% Add bibliography
\addbibresource{references.bib}

% Define Rust lisintgs
\lstset{
    language=Rust,
    showstringspaces=false
}
\DeclareFloatingEnvironment[
  name=Listing,
  listname=List of Listings,
  within=section,
]{listing}

\restylefloat{table}

% Remove indent across the whole document
\setlength\parindent{0pt}

% Define equation numbering
\numberwithin{equation}{section}

\newcommand{\beginsupplement}{%
  \renewcommand{\thesection}{}
  \setcounter{subsection}{0}
  \renewcommand{\thesubsection}{S\arabic{subsection}}
  \setcounter{table}{0}
  \renewcommand{\thetable}{S\arabic{table}}%
  \setcounter{figure}{0}
  \renewcommand{\thefigure}{S\arabic{figure}}%
  \setcounter{listing}{0}
  \renewcommand{\thelisting}{S\arabic{listing}}
}

\newcommand{\mbP}{\textbf{P: }}
\newcommand{\mbC}{\textbf{C: }}
\newcommand{\mbx}{\mathbf{x}}
\newcommand{\mbm}{\mathbf{m}}
\newcommand{\mbp}{\mathbf{p}}
\newcommand{\mbu}{\mathbf{u}}
\newcommand{\mbF}{\mathbf{F}}
\newcommand{\mbH}{\mathbf{H}}
\newcommand{\cmtb}[1]{\textcolor{blue} {#1}}
\newcommand{\cmt}[1]{\textcolor{red} {[#1]}}
\newcommand{\etal}{{\textit{et al. }}}

\newacro{ode}[ODE]{Ordinary Differential Equation}
\newacro{pde}[PDE]{Partial Differential Equation}
\newacro{glv}[gLV]{generalized Lotka-Volterra}
\newacro{abm}[ABM]{Agent-Based Model}


% Define new style for json listings
% Copied from https://www.latex4technics.com/?note=3QTU
\colorlet{punct}{red!60!black}
\definecolor{background}{HTML}{EEEEEE}
\definecolor{delim}{RGB}{20,105,176}
\colorlet{numb}{magenta!60!black}

\lstdefinelanguage{json}{
    basicstyle=\normalfont\ttfamily,
    numbers=left,
    numberstyle=\scriptsize,
    stepnumber=1,
    numbersep=8pt,
    showstringspaces=false,
    breaklines=true,
    frame=lines,
    backgroundcolor=\color{background},
    literate=
     *{0}{{{\color{numb}0}}}{1}
      {1}{{{\color{numb}1}}}{1}
      {2}{{{\color{numb}2}}}{1}
      {3}{{{\color{numb}3}}}{1}
      {4}{{{\color{numb}4}}}{1}
      {5}{{{\color{numb}5}}}{1}
      {6}{{{\color{numb}6}}}{1}
      {7}{{{\color{numb}7}}}{1}
      {8}{{{\color{numb}8}}}{1}
      {9}{{{\color{numb}9}}}{1}
      {:}{{{\color{punct}{:}}}}{1}
      {,}{{{\color{punct}{,}}}}{1}
      {\{}{{{\color{delim}{\{}}}}{1}
      {\}}{{{\color{delim}{\}}}}}{1}
      {[}{{{\color{delim}{[}}}}{1}
      {]}{{{\color{delim}{]}}}}{1},
}

\title{Pool Model - A unified Mathematical\\ Modelling Framework for Food Science}
\author{
    Polina Gaindrik\textsuperscript{1},
    Jonas Pleyer\textsuperscript{1}
    and Christian Fleck\textsuperscript{1}
}

\date{
    \footnotesize
    \textsuperscript{\textbf{1}}Freiburg Center for Data Analysis and Modeling
}

\begin{document}
\maketitle

\begin{abstract}
    Bacterial consortia are complex systems which are often comprised of multiple distinct species.
    Their dynamics on a population-based level often result from interactions that lead to
    inter-species cooperation or competition.
    Various empirical and inactivation models have been conceived and applied to problems related to
    food microbiology.
    However, the utilized models live on the popluation level, omitting any spatial effects such as
    heterogeneity or diffusion.
    In this work we present a novel generalized framework, called "pool model" which uses multiple
    distinct pools that can hold bacterial states of distinct species or extracellular nutrient
    abundance.
    It allows us to describe these systems mechanistically as variations of an underlying, more
    fundamental model.
    To illustrate the usefulness of the model, multiple different examples are reported.
    Finally, we construct an \acl{abm} for a given pool model, showcasing that the resulting model
    arises naturally from the individual behaviour of many cellular agents.
    We further apply this \ac{abm} to study the role of spatial effects on the dynamcis of bacterial
    growth.
    % \textbf{Notes from Christian}
    % Different empirical and inactivation models applied in the estimation
    % of parameters and modelling of phenomena in food microbiology, gen-
    % eralized Lotka Voltera approach used to describe the interaction of mi-
    % croorganisms, mechanistic models utilized to provide more insight into
    % the underlying dynamic can all be unified as variations of a new model
    % known as three pool model. Compared against the aforementioned
    % methods which are applicable in specific applications with some limi-
    % tations, the present model turns out to be less complex with general
    % applicability, low number of parameters and preferably to explain the
    % mechanism of the phenomenon of interest. To illustrate the generality
    % of the model, three different examples are reported.
\end{abstract}

\textbf{Keywords:}
Bacterial Consortia, Competition and Cooperation, Mechanistic Modeling, Spatial Effects

\newpage
\tableofcontents

\twocolumn

\newpage
%
%
%
%
% ================================================================================================
\section{Introduction}

% \paragraph{General Introduction to Bacterial Consortia}
% \begin{itemize}
%     \item Complex system; many players (cell-types, interactions)
%     \item processes on different scales (population, cellular, molecular, gene-level)
%     \item Food Safety, Production
%     \item Healthcare: Drug development
%     \item Environmental Sustainabiliy
% \end{itemize}
% Motivation: general, flexible approach for bacterial systems that allows easily modify the system depending on processes on different scales.
% Allows us to describe the processes within populations using different bacterial states as well as interactions between different populations using ODEs.

Bacterial consortia are highly complex systems which consist of multiple distinct cell types.
The underlying processes are interconnected with each other and occur on all scales, ranging from
population-based dynamics of colony growth over molecular processes such as metabolism to gene
expression.
Bacterial communities are ubiquitous and even able to survive under extreme conditions, making them
virtually omnipresent around any living system.
They play an especially important role within the context of food safety and production
being a key contributor to the spoilage of food products as well as fermentation processes.
Furthermore, effects such as biofilm formation can hinder drug treatment by shielding the bacterial
colony from antibiotic substances or immune response of the organism.
The natural bacterial processes are a crucial part of our planet's ecosystem, being responsible for
tasks such as nutrient cycling, decomposition of organic matter and oxygen production.
It is of great interest to understand the dynamics of these various processes.

% \paragraph{Experimental/Biological evidence/phenomena}
% \begin{itemize}
%     \item Lag Phase, Competition, Cooperation, Limiting Resource
%     \item Comprehensive reviews about competition/cooperation bacterial mechanisms: \cite{ghoul_ecology_2016, stubbendieck_bacterial_2016, hibbing_bacterial_2010, west_social_2007}
%     \item Spatial Effects: adhesion, migration, chemotaxis, diffusion
%     \item Quorum-sensing
%     \item Biofilm formation \cite{Jin2020}
% \end{itemize}

In order to achieve a comprehensive description of the target system, it is important to incorporate
all relevant biological phenomena arising in these bacterial systems.
During their reproductive process, bacteria undergo multiple phases~\cite{buchanan_when_1997}, starting in a lag phase which
is necessary to adapt to new environments and initially impedes the overall growth~\cite{rolfe_lag_2012, monod_growth_1949}.
Additionally, bacteria are able to cooperate e.g. via complementary metabolic processes or inhibit
each other e.g. by means of waste production as a metabolic byproduct~\cite{west_social_2007, rainey_evolution_2003, hibbing_bacterial_2010, stubbendieck_bacterial_2016}.
A limited supply of nutrients will also affect the overall reproductive capacity of the colony~\cite{monod_growth_1949}.
Bacteria have thus devised various survival strategies to ensure sufficient access to nutrients.
Effects on a single-cell level such as adhesion~\cite{htuson_bacteriasurface_2013}, migration~\cite{decoene_microscopic_2011} and chemotaxis can lead to complex
phenomena on a population-based level which can aid in the goal of survival.
Cell signaling e.g. in the form of quorum-sensing can lead to collective emergent phenomena such as
biofilm formation which play an important role for many mature microbial communities~\cite{ng_bacterial_2009}.

% \paragraph{ODE models (predictive microbiology, existing, overview, most popular)}
% Existing models in Predictive Microbiology (Primary Models)~\cite{perez-rodriguez_predictive_2012}:
% \begin{itemize}
%     \item An overview of primary models used in food science can be found in~\cite{van_boekel_kinetic_2008}.
%     \item Baranyi-Roberts Model~\cite{baranyi_modeling_1993, baranyi_dynamic_1994}
%     \item modified Gompertz~\cite{Gompertz1825-wi,Gibson1987}
%     \item Buchanan et al.: three-phase linear model for lag phase, exponential growth phase, and stationary phase~\cite{buchanan_when_1997}
%     \item McKellar two-compartment model with bacteria in growing and non-growing state~\cite{mckellar_heterogeneous_1997}
%     \item Inactivation models
% \end{itemize}
% %
% Secondary models:
% \begin{itemize}
%     \item Polynomial models 
%     \item Square Root-Type Models, e.g., Ratkowski for temperature~\cite{ratkowsky_relationship_1982}
%     \item The Cardinal Parameter Model~\cite{zwietering_decision_1992}
% \end{itemize}

The first step in the construction of any mathematical model is to determine the required level of
detail with respect to the description scale and consequently which effects to include.
The bacterial growth curves let us describe the dynamics of the overall abundance of various
species.
The field of predictive microbiology has developed a variety of ODE-based models to capture
bacterial growth dynamics~\cite{perez-rodriguez_predictive_2012}.
Various \acp{ode} \textit{primary models} describe the bacterial growth curve under constant
environmental conditions.
Widely used examples include the modified Gompertz model~\cite{Gompertz1825-wi}, Baranyi-Roberts
model~\cite{baranyi_modeling_1993, baranyi_dynamic_1994} and McKellar two-compartment model~\cite{buchanan_when_1997} which distinguishes between growing and non-growing bacteria.
To take into account changing environmental conditions, so-called \textit{secondary models}
formulate the model parameters as a function of external quantities such as temperature, pH or
humidity.
These can be simple polynomial models, square root-type models~\cite{ratkowsky_relationship_1982} or
more complex relations such as the cardinal parameter model~\cite{zwietering_decision_1992}.
The advantage of the \ac{ode} formulation lies in the predictive power and simplicity of usage.
However, due to their coarse-grained nature, they lack the ability to describe heterogeneity and
other spatial effects.

% \paragraph{Models for spatial effects; PDEs, \acp{abm}, others}
% \begin{itemize}
%     \item BSim \cite{Gorochowski2012,Dang2020}
%     \item Bacterial Branching patterns \cite{Kawasaki1997,Matsushita1998}
% \end{itemize}

% To incorporate spatial heterogeneity or other effects other modeling approaches have been devised.
% \acp{pde} follow a similar pattern as \acp{ode} but also distribute the described variables through
% space.
When dealing with spatial effects such as heterogeneity, we are required to use more extensive
modeling formulations such as \acp{pde} or \acp{abm}.
The former represent an intermediate step between the population-based \ac{ode} models and the
individually-based \acp{abm} and can be effective, especially when dealing with large crowds of
cells that can be approximated as a continuum.
On the other hand, \acp{abm} excel at describing the individual behaviour of cells which can then
lead to more complex collective phenomena.
Tools such as BSim~\cite{Gorochowski2012,Dang2020} or \texttt{cellular\_raza}~\cite{Pleyer2025} are
capable to describe large amounts of cells, performing various intracellular and extracellular
processes.
The latter has been used to adapt \ac{pde}-based models into an \ac{abm} formulation which describe
bacterial branching patterns~\cite{Kawasaki1997,Matsushita1998}.

% \paragraph{What is the problem/what is the question? How do we address this?}

In this work, we introduce a new \ac{ode}-based modeling framework which describes different
bacterial species, their metabolic states and external quantities such as nutrients as individual
pools.
The dynamics of bacterial growth can be understood in terms of transitions between the different
bacterial states i.e. pools.
Our model also allows us to capture interactions between these bacterial states and the external
microenvironment.
Compared against the aforementioned methods, this approach turns out to be less complex with general
applicability, low number of parameters and greater explainability for the underlying biological
reality.
To illustrate the generality of the model, multiple examples are reported.
We analyze the spatial limitations of an interacting two-species model and show that it can be fully
understood on an individual-based level in terms of single-cell processes.
Furthermore, we assess how spatial heterogeneity can impact the shape of the growth curves.


% ================================================================================================
\section{One Species Modeling}
% ------------------------------------------------------------------------------------------------
\subsection{Three-Pool Model}
Consider for simplicity the system where only one bacteria species is present $N=1$.
The bacterial growth curves usually consist of different consecutive phases~\cite{buchanan_when_1997}.
At the beginning no increase in bacterial count is present and so-called lag-phase is observed.
This phenomenon appears due to adaptation of the microbial community, for example, to the new environmental conditions.
This so-called preparation to the exponential growth includes synthesis of the needed compounds, initiation of the transcription~\cite{rolfe_lag_2012} and poorly studied physiological and regulatory mechanisms~\cite{monod_growth_1949}.
After this, the succeeded in adaptation bacteria enter the exponential growth phase, where the process of cell division is directly happening.
When the nutrition is exhausted by the growing bacteria, the stationary state of the growth curve occurs.
In this state the cultures accumulate waste products of metabolic activity that leads to cell death of the bacteria.
At the same time dying cells release the nutrients for survivors to continue growing.
As a result the balance between the growing and dying processes leads to the constant with time bacterial count~\cite{navarro_llorens_stationary_2010}.
In the described simplified version of the real-life system three possible bacteria states are mentioned corresponding to which process is happening: adaptation, growth or death.
Motivated by this we divide a bacterial population into three pools (states)~(\ref{fig:SchematicRep}) defining the fraction of the population in the lag phase $L(t)$, growing and dividing $G(t)$ and undergoing cell death $D(t)$.
\begin{figure}[t]
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures/TPM_fig.pdf}
    \caption{Schematic representation of exchange between three states or pools.}
    \label{fig:SchematicRep}
    \end{center}
\end{figure}
To determine the system of equations describing the system dynamics consider the flows between different states.
The pools $G$ and $D$ are unidirectionally connected with the pool $L$.
For now, we assume for simplicity that there is no back flow between growth state $G$ and lag-state $L$.
The suggested model can be cast in the following reactions:
\begin{eqnarray}
    L &\stackrel{\lambda}{\longrightarrow} & G\\
    L &\stackrel{\mu}{\longrightarrow} & D\\
    G &\stackrel{\alpha}{\longrightarrow} & 2G\\
    G &\stackrel{\mu'}{\longrightarrow} & D\\
    D &\stackrel{\beta}{\longrightarrow} & \emptyset
\end{eqnarray}
Then the \acp{ode} describing such a model can be written as
\begin{align}\begin{split}
    \dot{L} &= -(\mu + \lambda) L\\
    \dot{G} &= \lambda L + \alpha G - \mu' G\\
    \dot{D} &= \mu  L + \mu' G- \beta D  
\end{split}\end{align}.
%
Based on experimental observations, it was shown that the bacterial growth is strongly influenced by the limiting resource.
According to the works of Jacques Monod, in the system where all needed nutrients are in excess except for one the total growth is linear to the concentration of this limiting nutrient~\cite{monod_growth_1949}.
The control of the population density with respect to the available nutrition is implemented using cell communication process called quorum sensing.
Using release of the extracellular signalling molecules (autoinducer molecules) it is possible to synchronize the gene expression of the whole group~\cite{ng_bacterial_2009}.
We would not dive into the processes of quorum sensing and try to keep it as simple as possible.
According to this, we aim to include a resource pool which is used by the growing bacteria using the second order kinetic reaction:
\begin{equation}
    G + R_0  \stackrel{\alpha_0}{\longrightarrow} 2G.
\end{equation}
Note that the pools $L$, $G$ and $D$ represent the number of bacteria (or number of bacteria per unit area/volume) while $R_0$ represents an abstract resource pool.
Let say $G=N_B/V$ and $[G]=1/V$, where $N_B$ is the number of bacteria and $V$ is the volume.
Then $R_0=N_R/V$ and $[R_0]=1/V$, where $N_R$ is the number of resource molecules (or any appropriate unit, e.g., mol).
Then $[\alpha_0]=V/s$, $[N_t]=1/V$, and $[\alpha]=1/s$.
For homogenous densities it is equivalent to consider the total amount, i.e., multiplying $L$ and $G$ by the volume.
In this case $[L]=[G]=[N_t]=1$.
For simplicity, we rather make a transition to the dimensionless constant $\alpha:=\alpha_0 N_t$ and dimensionless resource pool $R(t) = R_0 / N_t$ ($0 \leqslant R \leqslant 1$).
Here the full capacity of the resource pool $N_t$ was used as a scaling constant.
In this case the growth term can be simply written as $\alpha G R$ and the model is fully formulated in the following way:
\begin{align}
    \begin{split}
        \dot{L} &= -(\mu + \lambda) L\\
        \dot{G} &= \lambda L + \alpha R G-\mu' G\\
        \dot{D} &= \mu  L + \mu' G- \beta D\\
        \dot{R} &= - \frac{\alpha}{N_t} R G
    \end{split}.
\label{eq:ode_3pool_resource} 
\end{align}
%
Such model can be simplified by exploiting that limitation of the resource pool also limits the total bacterial count of the system:
\begin{equation}
    R = 1 - \frac{L+G+D}{N_t}
\end{equation}
and, hence, the full system can be written as
\begin{align}
    \begin{split}
        \dot{L} &= -(\mu + \lambda) L\\
        \dot{G} &= \lambda L + \alpha G\left(1-\frac{L+G+D}{N_t}\right)-\mu' G\\
        \dot{D} &= \mu  L + \mu' G- \beta D 
    \end{split}.
\label{eq:ode_3pool_resource2} 
\end{align}
%
%
\begin{figure}
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_3pools_resource.pdf}
    \caption{
        The time dependence of the total bacterial count $N(t) = L(t) + G(t) + D(t)$ is obtained as a sum of bacteria in all states.
        All curves were calculated using the model (\ref{eq:ode_3pool_resource2}) with parameters $\alpha=2.5$, $\mu'=0.5$ and $\beta=5$.
        The blue curve shows significant dip in behavior of the system where the death of the bacteria is present $\mu = 0.5$.
        Other curves show the case of no bacteria dying $\mu=0$ and represent different length of lag phases:
        short $\lambda=10^{-2}$ (dark green), intermediate $\lambda=10^{-3}$ (light green) and long one $\lambda=10^{-4}$ (yellow).
    }
    \label{fig:3pool_resource_plots}
    \end{center}
\end{figure}
The resulting growth curves for the three pool case are presented in Figure~\ref{fig:3pool_resource_plots}.
Here we can see that the value of the transition constant $\lambda$ determines how long the bacteria is in lag-phase.
The larger is the rate the shorter is the lag-time.
Moreover, by including the bacteria flow to the death pool we can diversify the shape of the growth curve allowing the decrease of the bacterial count.
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Lag-time Estimation}
Note that one could further simplify this model to a two pool or state model by omitting the pool $D$ and removing cells with rate $\beta$ directly from pool $L$.
The relevant question here is whether cells in the pool $D$ still contribute to the limitation of the resources by, e.g., occupying space or consuming important nutrients.
If yes, the three pool model would be advantageous, if not the two pool model is sufficient with the advantages of having one parameter less.\\
%
The dynamics of the three pool model can be captures by the coupled system of ordinary differential equations (the two pool model can be achieved by setting $D\equiv 0$ and replacing $\mu$ with $\beta$):
to be solved with the initial conditions $L(0)=N_0$, $G(0)=0$, and $D(0)=0$, where $N_0$ is the initial bacterial population.
$N_t$ denotes the maximal size of the total bacterial population due to environmental limitations (Figure \ref{fig:3pool_resource_plots}).
For $\mu'=0$ the equations for $L$ and $D$ can be readily integrated resulting in:
\begin{align}
    \begin{split}
        L(t) &= N_0 e^{-(\mu+\lambda)t}\\
        D(t) &= N_0 \mu e^{-\beta t}\frac{1-e^{-(\mu+\lambda-\beta)t}}{\mu+\lambda-\beta}
    \end{split}.
\end{align}
For constant parameters $\mu$, $\lambda$, and $\beta$ the integrals for $L$ and $D$ can be put into the differential equation for $G$.
However, for time dependent parameters, e.g., for a dependence on a non-static, dynamic temperature, it appears to be not advantageous to use the analytical results for $L$ and $D$ and work instead with all three equations for $L$, $G$, and $D$.
It is interesting to see what type of equations result for the total bacterial population $N=L+G+D$:
\begin{align}
    \begin{split}
        \dot{G} &= \lambda L + \alpha G\left(1-\frac{N}{N_t}\right)\\
        \dot{N} &= - \beta D + \alpha G\left(1-\frac{N}{N_t}\right)
    \end{split}
\end{align}
with the initial conditions $G(0)=0$ and $N(0)=N_0$. 
These equations have a clear interpretation, are derived from simple principles, and are different from the popular Barayni-Roberts and modified-Gompertz models.
A notable difference is that the growth rate $\alpha$ is time independent.
For the functions $L$ and $D$ it holds: $\lim_{t\to\infty} L(t) = \lim_{t\to\infty} D(t) = 0$.
It follows immediately that $\lim_{t\to\infty} G(t) = N_t$.
The equation exhibit a lag phase and an initial drop due to cell death.
To find a substantial initial drop one needs $\beta \mu \gg \alpha \lambda$.
The lag time $t_L$ can be approximated by the time at which half of the initial population is in the growth phase, i.e., $N_0=2G(t_L)$.
Assuming $L\ll N_t$ we can ignore the non-linear term in the equation for $G$, which gives rise to:
\begin{align}
    1&= 2\int_0^{t_L} \lambda L(t)e^{\alpha(t_L-t)}dt\\
    1  &= 2\lambda e^{\alpha t_L}\frac{1-e^{-(\mu+\lambda+\alpha)t_L}}{\mu+\lambda+\alpha}.
\end{align}
For $\alpha > \mu+\lambda$ $t_L$ can be approximated by:
\begin{align}
    t_L &\approx -\frac{1}{\alpha}\ln\left(\frac{2\lambda}{\mu+\lambda+\alpha}\right).
\end{align}
The lag time $t_L$ diverges very slowly, logarithmically with $\lambda\to 0$.
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Back flow from Growth to Lag pool}
Due to its life in rather unfavorable conditions bacteria developed different surviving mechanisms, one of which is adapting their metabolism and transitioning into dormant state.
By definition dormancy in bacteria is a reversible state where cell continue living with low metabolic activity while the cell division stops~\cite{kaprelyants_dormancy_1993}.
This survival strategy can appear in response to different external stresses, e.g., nutrient deficiency, temperature stress or extreme fluctuations of environment parameters.
And after "difficult times" bacteria, also called viable but non-culturable (VBNC), can continue cell division~\cite{kell_viability_1998}.
The models including transition of bacteria to the dormant state were proposed and studied by Bär \etal~\cite{bar_modelling_2002} and Jones and Lennon~\cite{jones_dormancy_2010}.
%
We explore now the idea that under certain conditions, as sudden change in the environment, part of the population $G$ will enter a lag-phase, i.e., there is a back-flow from $G$ to $L$.
\begin{eqnarray}
    L &\stackrel{\lambda}{\longrightarrow} & G\\
    L &\stackrel{\mu}{\longrightarrow} & D\\
    G &\stackrel{\alpha}{\longrightarrow} & 2G\\
    G &\stackrel{\gamma(t)}{\longrightarrow} & L\\
    D &\stackrel{\beta}{\longrightarrow} & \emptyset
\end{eqnarray}
The rate $\gamma(t)$ depends on the rate of change of the environment.
Stronger and quicker changes will lead to a higher rate $\gamma$.
%
%
\paragraph{Maxwell type of stress - strain relation}
One of the reasons of the dormant state appearance can be also the temperature shift~\cite{oliver_viable_1995}.
Moreover, the rapid shift in temperature is associated with the appearance of an additional lag phase~\cite{zwietering_modeling_1994}.
To model it we propose a kind of viscoelastic Maxwell type of stress-strain relation and write down the ordinary differential equation for $\gamma$ to be:
\begin{equation}
    \dot{\gamma} = \Gamma |\dot{T}|-\delta \gamma.
\end{equation}
We assumed that the direction of the temperature change does not matter, i.e., a change from $T=2$ to $T=12$ has the same effect as a change from $T=14$ to $T=4$.
$\Gamma$ is a scaling factor and $\delta$ is the relaxation time for the temperature disturbance.
Integration of this equation yields (with $\gamma(0)=0$):
\begin{equation}
    \gamma(t) = \Gamma \int_0^t |\dot{T}(t^{\prime})|e^{-\delta (t-t^{\prime})}dt^{\prime}.
\end{equation}
After some manipulation of this equation we find:
\begin{equation}
    \gamma(t) = \Gamma\delta\left |T[t]-T[0]e^{-\delta t}-\delta
        \int\limits_0^t T(t^{\prime})e^{-\delta (t-t^{\prime})}dt^{\prime}\right |.
\end{equation}
This representation of the rate $\gamma(t)$ has the advantage that it only involves $T(t)$ and does not require the calculation of the derivative of $T$.
If $T$ exhibits $n$ temperature jumps, one finds:
\begin{eqnarray}
    \gamma(t) &=& \Gamma\sum_{i=1}^n \left |\Delta T_i \right |\theta(t-t_i)e^{-\delta(t-t_i)}.
\label{eq:gamma_tempshift}
\end{eqnarray}
%
The ordinary differential equations for the pool model with environmental-shock based back-flow read:
\begin{align}
    \begin{split}
        \dot{L} &=\gamma(t) G - (\mu + \lambda) L\\
        \dot{G} &= \lambda L -\gamma(t) G + \alpha G\left(1-\frac{L+G+D}{N_t}\right)\\
        \dot{D} &= \mu  L - \beta D
    \end{split}
\label{eq:ode_tempshift_backlag}
\end{align}
\begin{figure}[t]
    \begin{center}
    %\includegraphics[width=0.9\columnwidth]{Figures/pool_model_3pools_resource_tempshift.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_3pools_resource_tempshift1.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_3pools_resource_tempshift2.pdf}
    \caption{
        (a) The time dependence of the total bacterial count $N$ for the model including the back-flow to the lag phase due to temperature stress~(\ref{eq:ode_tempshift_backlag}).
        (b) The value of the corresponding back-flow rate $\gamma$ calculated according to (\ref{eq:gamma_tempshift}).
        The curves represent no temperature shift (green), one temperature jump $\Delta T = 10$ at time $t=4$ (orange)
        and the series of different temperature shifts of value 10, 5 and 15 at the time points 1, 3 and 7, respectively (rose).
        The parameters of the back-flow rate are $\Gamma=1$ and $\delta=5$.
        The other model parameters are $\alpha=2.5$, $\mu'=0.5$, $\beta=5$, $\lambda=10^{-2}$, $\mu = 0$ for each curve.
    }
    \label{fig:TempJump}
    \end{center}
\end{figure}
The effect of the temperature shift on the total bacterial count can be shown in Figure~\ref{fig:TempJump}.
After the rapid temperature increase, the growth slows down till the bacteria that transitioned back to lag-phase go again to growth pool.
If the shift happened rather early in time, when most of the bacteria were in lag-phase, the effect is not noticeable.
%
% ================================================================================================
\section{Interaction between species}
As it was mentioned above, bacteria usually coexist in large multispecies interacting communities whose growth is vastly determined by the developed survival mechanisms.
These survival mechanisms mostly aim to secure as much available nutrition which lead to competitive strategies or, if it is advantageous, cooperative ones~\cite{hibbing_bacterial_2010, stubbendieck_bacterial_2016}.
Further we show that the model can be readily extended to the systems with more than one bacterium and describe more complex relationships between species. 
%
% ------------------------------------------------------------------------------------------------
\subsection{Resource Competition}
One on the ways to describe competition between bacteria populations is as a function of limiting resource ratio~\cite{tilman_resource_1977, smith_effects_2002}.
To describe such a situation we consider for simplicity only one common limiting resource pool.
The simplest pool model for bacterial species $A$ and $B$ encompassing lag-phases, growth-phases and a common resource pool is given by (no back-flow to the lag-pool):
\begin{align}
    \begin{split}
        \dot{L}_A &= - \lambda_A R L_A\\
        \dot{L}_B &= - \lambda_B R L_B \\
        \dot{G}_A &= \lambda_A R L_A +\alpha_A R G_A\\
        \dot{G}_B &= \lambda_B R L_B +\alpha_B R G_B\\
        \dot{R} &=-\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{N_t} R G_B
    \end{split}
    \label{eq:model_2sp_resource_comp}
\end{align}
We again rescaled the resource pool with $N_t$ and find
\begin{equation}
    R =1-\frac{L_A+G_A+L_B+G_B}{N_t}
    \label{eq:Resource}
\end{equation}
Inserting Eq.~\ref{eq:Resource} into Eqs.~\ref{eq:model_2sp_resource_comp} yields:
\begin{align}
    \dot{L}_A &= -\lambda_A  L_A + \frac{\lambda_A}{N_t}L_A\left<LG\right>\\
    \dot{L}_B &= -\lambda_B L_B + \frac{\lambda_B}{N_t}L_B \left<LG\right>\\
    \dot{G}_A &=  \lambda_A  L_A + \alpha_A G_A - \frac{\lambda_A L_A + \alpha_A G_A}{N_t} \left<LG\right>\\
    \dot{G}_B &=  \lambda_B L_B - \alpha_B G_B \frac{\lambda_B L_B + \alpha_B G_B}{N_t}\left<LG\right>
\end{align}
where we used the short notation $\left<LG\right>=(L_A+G_A+L_B+G_B)$.
These equations have the structure of a \ac{glv} expanded with different internal states for one species.
As one can see in Figure~\ref{fig:2pool_resource_2sp} even without direct interaction between species the bacterial growth was inhibited due to the limitation of the resource.
Here species $A$ has a longer lag-phase than $B$ but the exponential growth of $A$ is stronger than the one that bacteria $B$ experiences.
As a result, species $A$ was able to suppress species $B$ consuming the larger amount of nutrient and reaching larger bacterial count at the end.
\begin{figure}
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_2pools_resource_competition.pdf}
    \caption{
        The competition of the two bacterial species due to the common limiting resource calculated using Eq.~(\ref{eq:model_2sp_resource_comp}).
        The red curve describes the growth of the bacteria $A$ ($N_A = L_A+G_A$) with model parameters $\lambda_A=10^{-2}$, $\alpha_A=3.0$.
        The blue curve describes the growth of the bacteria $B$ ($N_B = L_B+G_B$) with model parameters $\lambda_B=5\cdot 10^{-2}$, $\alpha_B=2.5$.
        The purple line shows the total bacterial count of the system  $N(t)=N_A+N_B$.
        The gray line denotes the dynamic of the resource pool scaled with its capacity $N_t=10^4$.
    }
    \label{fig:2pool_resource_2sp}
    \end{center}
\end{figure}
%
In order to get the commonly used form of the \ac{glv}, we need to simply eliminate of the bacterial states by setting $L_A=L_B=0$:
\begin{align}
    \begin{split}
        \dot{G}_A &= \alpha_A G_A\left(1 - \frac{G_A}{N_t}\right) - \frac{\alpha_A}{N_t}G_AG_B\\
        \dot{G}_B &= \alpha_B G_B\left(1-\frac{G_B}{N_t}\right) -\frac{\alpha_B}{N_t}G_AG_B. 
    \label{eq:LV_simple}
    \end{split}
\end{align}

Equations (\ref{eq:LV_simple}) are basically competitive Lotka-Volterra equations that are widely used in ecological studies and show the principle of competitive excursion.
That principle claims that two species in competition for the same resource cannot coexist~\cite{wangersky_lotka-volterra_1978}.
In previous studies it was shown that \ac{glv} equations with one bacterial state can be successfully applied to predict the microbial community dynamics, determine different types of the pairwise interspecies interactions~\cite{dedrick_when_2023, stein_ecological_2013, venturelli_deciphering_2018, hoffmann_power_2007}.
In food studies the \ac{glv} was applied by Mounier \etal~\cite{mounier_microbial_2008} to study interactions of six bacteria species influenced by yeast omissions in cheese.
Cauchie \etal~\cite{cauchie_modeling_2020} applied Lotka-Volterra model to study the three-species system of the bacteria majorly present in minced pork meat.\\
%
However, in contrast to the classical Lotka-Volterra approach, the introduced pool model allows for far more flexibility in tuning properties of the bacterial interactions within and between species.
For further comparison of the derived model with the matrix formulation of the \ac{glv} we refer the reader to the supplement materials~\ref{ssec:supplement1}.
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Interspecies Competition}
However, except the inhibition due to resource competition, bacteria developed different others active competition mechanisms.
One of the examples of such mechanisms is a production of the antimicrobial compounds, inhibitors or toxins~\cite{wloch-salamon_effect_2008, chao_structured_1981}.
On the other hand, some bacteria can produce public goods, the compounds that benefit growth of other species.
In this way, can suppress the non-preferred species by helping their competitors~\cite{hibbing_bacterial_2010}.
%
%
% ................................................................................................
\subsubsection{Toxins Production}
To introduce active competitive mechanisms, we extend our model with one of the species producing the toxic compounds $T$ during its growth:
\begin{equation}
    \dot{T} = \mathcal{F}(R,G).
\end{equation}
The function $\mathcal{F}$ denotes the model for the toxin production rate.
It can be, for example, the accumulation of the bacteria waste which is responsible for the spoilage of food products and affecting another species in negative way.
Consider the system for two species competing for resource $R$ with the production of the toxin $T$ by species $B$.
Assume that the toxin causes the death of the species $A$ with rate $\mu$.
Then the full \ac{ode} system is
\begin{align}
    \begin{split}
        \dot{L}_A &= - \lambda_A R L_A\\
        \dot{L}_B &= - \lambda_B R L_B \\
        \dot{G}_A &= \lambda_A R L_A + \alpha_A R G_A - \frac{\mu}{\kappa N_t} T G_A\\
        \dot{G}_B &= \lambda_B R L_B + \alpha_B R G_B\\ 
        \dot{R} &= -\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{N_t} R G_B\\
        \dot{T}_{B}^{1,2} &= \mathcal{F}^{1,2} (R, G_B) \\
    \end{split}.
    \label{eq:model_2sp_toxin}
\end{align}
%
Where we consider two possible models of the toxin production:
\begin{align}
    \mathcal{F}^1(R,G)&=\kappa G_B\\
    \mathcal{F}^2(R,G)&=\kappa\alpha_B R G_B.
\end{align}
Here $\kappa$ is the toxin production rate.
The difference between the two toxin production models for resulting solution of the \acp{ode} is presented in Figure~\ref{fig:2pool_2sp_toxin}.
%
%
\begin{figure}
    \begin{center}
    %\includegraphics[width=\columnwidth]{Figures/pool_model_2pools_toxin.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_2pools_toxin1.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_2pools_toxin2.pdf}
    \caption{
        The two species system competing for the common nutrient $R$ with production of the toxin $T_B$ (brown line) calculated using Eq.~(\ref{eq:model_2sp_toxin}).
        The growing bacteria $B$ produces toxin $T_B$ (a) constantly $\dot{T}_B^1 = \mathcal{F}^1 = \kappa G_B$ or (b) only during growth process $\dot{T}_B^2 = \mathcal{F}^2 = \kappa \alpha_B R G_B$ with 
        production rate $\kappa=0.2$.
        The toxin $T_B$ causes the cell death of bacteria $A$ with rate constant $\mu = 0.2$.
        The red curve describes the growth of the bacteria $A$ ($N_A = L_A+G_A$) with model parameters $\lambda_A=10^{-2}$, $\alpha_A=3.0$.
        The blue curve describes the growth of the bacteria $B$ ($N_B = L_B+G_B$) with model parameters $\lambda_A=5\cdot 10^{-2}$, $\alpha_A=2.5$.
        The gray line denotes the dynamic of the resource pool scaled with its capacity $N_t=10^4$.
    }
    \label{fig:2pool_2sp_toxin}
    \end{center}
\end{figure}
%
Model $\mathcal{F}^1$ results in toxin concentration $T_B^1(t)=\kappa\int_{t_0}^tG(t')dt'$.
That means that as long as there are bacteria in the growth phase $G_B>0$ toxin is produced, no matter whether the total bacterial count $N_B$ is still growing.
On the other hand, model $\mathcal{F}^2$ results in toxin accumulation in steady state $T_B^{2s}=\lim_{t\to\infty}T_B^2(t)=\kappa(N_t-N_0)$.
To see this we note $\dot{T}_B^2=\kappa(\dot{L_B}+\dot{G_B})$, which yields: $T_B^2(t)=\kappa(L_B(t)+G_B(t)-N_{B0})$ where we used $T_B^2(0)=G_B(0)=0$, $L_B(0)=N_{B0}$.
This model assumes that toxin is only produced when the total bacterial grow and divide, hence, the total bacterial count $N_B$ increases.
%
%
% ................................................................................................
\subsubsection{Mutual Inhibition}
Another special case of the species competition is when the produced toxin does not cause the death of the bacteria but rather inhibit its growth.
Such situation with 2 bacteria species, limiting resource and produced inhibitor was modeled, e.g., by Lenski and Hattingh~\cite{lenski_coexistence_1986}.
In our case, we would like to look at rather simplified case reducing the number of variables of the system.
The easiest way to introduce inhibition of species $A$ by species $B$ is via a growth rate of species $A$ depending on the abundance of species $B$.
For sake of simplicity we ignore the lag pool and write for a mutual inhibition:
\begin{align}
    \dot{G}_A &= \mathcal{B}_A(G_B)G_A\left(1 - \frac{G_A+G_B}{N_t}\right)\\
    \dot{G}_B &= \mathcal{B}_B(G_A) G_B\left(1-\frac{G_A+G_B}{N_t}\right).
\end{align}
The function $\mathcal{B}_{A/B}$ captures the inhibiting effects effectively.
It is straight forward to make the mechanism of inhibition more explicit by, e.g., introducing a toxin pool.
One possible choice for $\mathcal{B}$ is:
\begin{equation}
    \mathcal{B}(G) = \frac{\alpha}{1+KI(G)},
\end{equation}
where $I$ is the concentration of the inhibiting metabolic byproduct producing during bacterial growth.
\begin{align}\begin{split}
    \dot{G}_A &= \frac{\alpha_A G_A}{1+K_BG_B}\left(1 - \frac{G_A+G_B}{N_t}\right)\\
    \dot{G}_B &= \frac{\alpha_B G_B}{1+K_AG_A}\left(1-\frac{G_A+G_B}{N_t}\right).
\end{split}\end{align}
Note that if the toxin accumulates, i.e., $\dot I \sim G$,  one needs to replace $G$ by $\int G dt$.
One can distinguish two limiting cases: {\it i}) $K_{A/B}\ll N_t$ and {\it ii}) $K_{A/B}\gg N_t$.
In the first case the inhibition is of minor importance and could be ignored.
In the latter case one can simplify the equations, given $N_A$ and $N_B$ are non-zero:
\begin{align}\begin{split}
    \dot{G}_A &=\tilde{\alpha}_A\frac{G_A}{G_B}\left(1 - G_A-G_B\right)\\
    \dot{G}_B &= \tilde{\alpha}_B\frac{G_B}{G_A}\left(1 - G_A-G_B\right).
\end{split}\end{align}
We introduced the rescaled growth rates $\tilde{\alpha}_A=\alpha_A/(N_tK_B)$ and $\tilde{\alpha}_B=\alpha_B/(N_tK_A)$. Note that $[\tilde{\alpha}]=\mathrm{s}^{-1}$.
Rescaling time with $\tilde{\alpha}_A$ and defining the dimensionless parameter
\begin{equation}
    \psi = \tilde{\alpha}_B/\tilde{\alpha}_A = \frac{K_B\alpha_B}{K_A\alpha_A}
\end{equation}
results in: 
\begin{align}
    \begin{split}
        \dot{G}_A &=\frac{G_A}{G_B}\left(1 - G_A-G_B\right)\\
        \dot{G}_B &=\psi\frac{G_B}{G_A}\left(1-G_A-G_B\right). 
    \end{split}
\label{eq:2sp_1pool_inhibit_unitless}
\end{align}
We can separate three cases (which are displayed in Figure~\ref{fig:1pool_2sp_inhibit}):
\begin{align}
    \psi&=1:\quad\text{A and B are equivalent}\\
    \psi&>1:\quad\text{B grows faster/inhibits A}\\
    \psi&<1:\quad\text{A grows faster/inhibits B}
\end{align}
%
\begin{figure}
    \begin{center}
    % TODO make this into 2x2 grid for plots and insert cases for psi from before as well
    %\includegraphics[width=\columnwidth]{Figures/pool_model_1pool_inhib.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_1pool_inhib1.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_1pool_inhib2.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_1pool_inhib3.pdf}
    \caption{
        The two-species system mutually inhibiting each other is defined with equations~(\ref{eq:2sp_1pool_inhibit_unitless}).
        Here the red lines describes the evolution of the species $A$ and blue lines correspond to species $B$.
        The relation between inhibition strength of species $B$ and $A$ are defined by control parameter value (a) $\psi$ = 1, (b) $\psi$ = 0.5 or (c) $\psi$ = 2.
    }
    \label{fig:1pool_2sp_inhibit}
    \end{center}
\end{figure}
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Interspecies Cooperation}
On the other hand, one can also observe some cooperative behaviors between species in case when it facilitates the competition with others.
It happens if bacteria species as a result of their metabolism produce the compounds that are beneficial for the growth of each other, so-called public goods.
Quite often it happens between the species that are close relatives or clonal populations and such process is called kin selection \cite{west_social_2007}.
Furthermore, neighbors can also benefit from a certain behavior without expressing it by itself, so-called 'social cheaters'~\cite{rainey_evolution_2003}.
%
%
The benefit, or mutual activation, of the two species from the presence of the other can be simply introduced by mathematical functions.
For sake of clarity we omit again the lag pool and write:
\begin{align}
    \dot{G}_A &= \mathcal{B}_A(P_B,R)G_A\\
    \dot{G}_B &= \mathcal{B}_B(P_A,R)G_B\\
    \dot{R} &=-\mathcal{B}_A(P_B,R)G_A-\mathcal{B}_B(P_A,R)G_B\\
    \dot{P}_A &= \mathcal{F}_A(R,G_A)\\
    \dot{P}_B &= \mathcal{F}_B(R,G_B).
\end{align}
$P_A$ and $P_B$ denote the substance produced by species A and B, respectively.
The functions $\mathcal{B}_A(P_B,R)$ and $\mathcal{B}_B(P_A,R)$ capture the resource dependent growth.
Setting $\mathcal{B}_A(P_B,R)=\alpha_A P_BR$, $\mathcal{B}_B(P_A,R)=\alpha_B P_A R$, $\mathcal{F}_A(R,G_A)=\kappa_A RG_A$, $\mathcal{F}_B(R,G_B)=\kappa_B RG_B$, rescaling $G_A$, $G_B$, and $P_B$ with $N_t$, $P_A$ with $\kappa_A/\alpha_A$, time with $\alpha_AN_t^2$, defining $\psi=\kappa_A\alpha_B/(\alpha_A^2N_t)$, and $\phi=\kappa_B/(\alpha_A N_t)$ results in:
\begin{align}
    \dot{G}_A &= P_B G_A\left(1 - G_A-G_B\right)\\
    \dot{G}_B &= \psi P_A G_B\left(1 - G_A-G_B\right)\\
    \dot{P}_A &= G_A\left(1 - G_A-G_B\right)\\
    \dot{P}_B &=\phi G_B\left(1 - G_A-G_B\right).
    \label{eq:1pool_2sp_cooper}
\end{align}
%
The solution of the system~\ref{eq:1pool_2sp_cooper} is presented in Figure~\ref{fig:1pool_2sp_cooper}.
Here we can see that if $\psi=1$ and $\phi=1$ (see Figure~\ref{fig:1pool_2sp_cooper}a) the production of the public goods of the two species is happening with the same rate as well as the growth of bacteria.
Increasing the production rate of the public good produced by species $B$ (see Figure~\ref{fig:1pool_2sp_cooper}b) causes increasing the growth of species $A$ that profit from the presence of $B$.
On the other hand, choosing $\psi=2$ (see Figure~\ref{fig:1pool_2sp_cooper}c), we increase the profit that species $B$ has from public goods $P_A$.
That indirectly leads to larger amount of $P_B$ produced by species $B$.
\begin{figure}
    \begin{center}
    %\includegraphics[width=0.9\columnwidth]{Figures/pool_model_2sp_1pool_coop.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_2sp_1pool_coop1.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_2sp_1pool_coop2.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures/pool_model_2sp_1pool_coop3.pdf}
    \caption{
        The two-species system where each of the species release public goods $P$ is defined with equations~(\ref{eq:1pool_2sp_cooper}) and calculated for different parameters:
        (a) $\psi=1, \phi=1$, (b) $\psi=1, \phi=2$ and (b) $\psi=2, \phi=1$.
        Here the red lines describes the evolution of the species $G_A$ and blue lines correspond to species $G_B$.
        Light brown curve corresponds to public good released by species $A$ ($P_A$) and brown curve shows public goods produced by species $B$ ($P_B$).
    }
    \label{fig:1pool_2sp_cooper}
    \end{center}
\end{figure}

% ================================================================================================
\section{Spatial Effects}
As described above, the proposed model operates with the total number of the bacterial counts for
different species and states in the system, not accounting for their spatial distributions.
Assuming that the bacterial system is uniformly distributed (sometimes also called 'well-stirred')
we expect the \ac{ode} description to be valid.
However, in reality spatial effects can play a significant role for population survival forcing some
species to develop competitive mechanisms such as cell adhesion to surfaces in the areas rich with
nutritients~\cite{htuson_bacteriasurface_2013}.
Another mechanism is the motility of bacteria, e.g., the flagellates bacteria as \textit{E. coli} or
\textit{B. subtilis} can experience the chemotactic motion of driven by oxygen
concentrations~\cite{decoene_microscopic_2011}.
The bacterial social interactions such as releasing the toxins or public goods can also
significantly affect the spatial structure of the community~\cite{blanchard_bacterial_2015}.
These effects can only be treated with a spatial model that is able to describe the observed
effects.\\
In order to construct a model which can encapsulate diffusion of nutrients and heterogeneity of
the bacteria or nutrients, we have to take into consideration the spatial component of our setup.
\acp{abm} force researchers to think in terms of cellular processes, thus bridging the gap from a
collective (bulk) model to an individual-based (single-cell) description.
We consider the system consisting from two bacterial species that can be described by two internal
states: lag and growth states.
Both of these species grow using the same resource pool $R$ but with different growth constants
$\alpha$ and different lengths of lag phase $\lambda$.
Moreover, the species $A$ is able to release a certain antimicrobial component $I$ that shows an
inhibitory effect on species 2 (amensalism).
We assume that $L,G$ are measured in terms of the overall volume and $R,I$ are measured as total
amount.
The \acp{ode} the bacterial population can be written as:
%
%
% - \cite{emonet_agentcell_2005, gorochowski_bsim_2012, li_nufeb_2019, kreft_bacsim_1998}: examples of another toolboxes for bacterial communities modeling (ABM).\\
% More about the modelling of the motile bacteria can be seen in works:~\cite{decoene_microscopic_2011, rosser_modelling_2014, sokolov_physical_2012, li_amplified_2008}.
%
%
% In general, bacterial communities in 2D can experience collective motion called swarming~\cite{wu_collective_2015}.
% Some works modelled 2D motile bacteria:
% \begin{itemize}
%     \item \cite{decoene_microscopic_2011}: 2D chemotactic self-propelled bacterial motion of the flagellated bacteria (e.g. E. coli, B. subtilis).
%     Chemotaxis here was driven by the oxygen concentrations.
%     \item \cite{rosser_modelling_2014}: modeling of the bacterial reorintation, rotation.
%     \item \cite{sokolov_physical_2012, li_amplified_2008}: modeling of the bacteria using hydrodynamic interactions and Brownian motion.
%     Fluctuations in velocity and rotations.
%     (\cite{li_amplified_2008}: Brownian motion gives right motion observed using microscopy).
%     \item \cite{farrell_mechanically_2013}: 2D modeling using Newtonian dynamics and only mechanical interactions + nutrient diffusion.
%     Branching parameter defines diff limits where diffusion dominant and limits growth, and where it's irrelevant.
%     Mechanical pressure (due to bacteria-bacteria interactions) affects growth, can limit it.(!!!)
% \end{itemize}
%
% For simulation results:
% \begin{itemize}
%     \item \cite{farrell_mechanically_2013}: Showed the influence of mechanic interactions between bacteria in 2D system with bacteria consuming the nutrient.
%     The growth of bacterial colonies where non-motile microorganisms replicate and push each other away as they grow.
%     Found a transition between two different growth regimes, controlled by the balance between growth and uptake of nutrient:\\
%     Exponential growth where diffusion of the nutrient does not play any role. Sublinear growth where growth is limited by nutrient diffusion.
%     (Structurally in 2D it's regimes: smooth propagating of bacterial front vs. branching.)
%     \item \cite{nagarajan_agent-based_2022-1}: review on the ABM for bacterial population.
%     The ABM represent a microbial colony as a group of discrete agents that are governed by a set of rules, where each agent can be an individual cell or a collection of cells.
%     Can use citation when talk about general processes we are accounting as growth, uptake of nutrient, production of inhibitor, e.g.,:
%     Physical interactions between agents are calculated using excluded volume interactions.
%     Chemical interactions as releasing inhibitors are described using concentrations and solved using PDEs.
% \end{itemize}
%
\begin{align}
    \dot{L_A} &= -\lambda_A R L_A\\
    \dot{G_A} &= \lambda_A R L_A + \alpha_A R G_A\\
    \dot{L_B} &= -\lambda_B R L_B\\
    \dot{G_B} &= \lambda_B R L_B + \frac{\alpha_B}{1 + \mu_I I} R G_B
    \label{eq:spatial_limit_F}
\end{align}
%
The microenvironment of our system contains the resource and the inhibitor molecules:
\begin{align}
    \dot{R} &= -\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{(1 + \mu_I I) N_t} R G_B \\
    \dot{I} &= \kappa G_A
    \label{eq:spatial_limit_H}
\end{align}

% ------------------------------------------------------------------------------------------------
\subsection{Constructing an Agent-Based Pool Model}
The following sections will demonstrate how a given \ac{ode} system can be translated into a
custom-built \ac{abm} which has only a few more overall parameters.
To construct our \ac{abm}, we use \texttt{cellular\_raza}~\cite{Pleyer2025} which was designed to
provide maximal flexibility for researchers in order to construct minimalist models.
Without the ability to define and control the precise behavior of our cell-agents, the following
approaches would not be possible.

\subsubsection{Cellular Reactions}
To construct an \ac{abm} which resembles the given \ac{ode} system of equations~\eqref{eq:spatial_limit_F} and \eqref{eq:spatial_limit_H}, we consider multiple cellular reactions.
The first is uptake of an extracellular resource $[R_e]$ into the cell $[R_i]$ with rate $u$ in \unit{\per\minute}.
Nutrients are only taken up at the position of the cell $x_c$
\begin{align}
    \dot{[R_i]} &= u [R_e](x_c)\\
    \dot{[R_e]}(x) &= -u \frac{V_c}{V_D} [R_e] \delta(x-x_c)
\end{align}
where $\delta(x-x_\text{cell})$ is the Dirac delta-distribution.
Note that $[R_e]=[R_e](x)$ explicitly depends on the spatial coordinate but we choose to omit its notation.
In the second equation, we need to account for the ratio of cellular $V_c$ and domain volume $V_D$ to calculate the correct change.
Further, we want to modify the cellular volume by growing it increasingly with rate $a$.
Therefore, we need to specify how many resource molecules are turned into one volumetric units $\sigma$ in \unit{\micro\metre\cubed\per\mole}.
The total amount of resource molecules present inside the cell is given by the current concentration multiplied by the cell-volume $V_c [R_i]$.
With a rate $a$ and conversion $\sigma$, the cell turns these resource molecules into volume.
\begin{align}
    \dot{V_c} &= a\sigma [R_i] V_c\\
    \dot{[R_i]} &= - a[R_i]
\end{align}
Furthermore, we let cells produce an inhibitor with production rate $b$ which inhibits the cell ability to uptake and thus use the resource by a factor of $1+\nu_I [I]$
\begin{align}
    \dot{[I]} &= b\frac{V_c}{V_D}\delta(x-x_c)\\
    u &\longrightarrow \frac{u}{1+\nu_I [I]}.
\end{align}
Note that we do not need to distinguish between the two species because we can set parameters individually on agent-level.
This means, we can implement the most general version of the equations and by choosing appropriate parameters, set the species of the respective cell-agent.
If we combine all equations and consider multiple cells, we obtain the system
\begin{align}
    \dot{V_c} &= a\sigma [R_i] V_c\\
    \dot{[R_i]} &= \frac{u}{1+\nu_I [I]} [R_e] - a[R_i]\\
    \dot{[R_e]} &= -\frac{u}{1+\nu_I [I]} \frac{V_c}{V_D} [R_e] \delta(x-x_c)\\
    \dot{[I]} &= b\frac{V_c}{V_D}\delta(x-x_c).
    \label{eq:spatial-full-system-non-simplified}
\end{align}
In order to simplify this system, we can assume that the internal resources are constantly being depleted by the growth and the limiting process is the uptake of the resource.
This means, we can effectively set $\dot{[R_i]}=0$.
Therefore, we obtain the relation $\frac{u}{1+\nu_I [I]}[R_e] = a[R_i]$ which we can use to eliminate the variable monitoring internal resource concentrations $[R_i]$.
Additionally, we require our external components to be freely diffusable with constants $D_R,D_I$.
Plugging this into equation~\eqref{eq:spatial-full-system-non-simplified} yields.
\begin{align}
    \dot{V_c} &= \frac{u}{1+\nu_I [I]} \sigma [R]V_c\\
    \dot{[R]} &= \sum\limits_c -\frac{u}{1+\nu_I [I]} \frac{V_c}{V_D} [R] \delta(x-x_c) + D_R\Delta [R]\\
    \dot{[I]} &= \sum\limits_c b\frac{V_c}{V_D}\delta(x-x_c) + D_I\Delta [I].
    \label{eq:spatial-full-system-simplified}
\end{align}
For the diffusion, we require Neumann (zero-flux) boundary conditions.
Since, we will not explicitly model intracellular concentrations $[R_i]$, we will drop the index entirely, only considering the extracellular resource.
%
%
% ................................................................................................
\subsubsection{Physical Mechanics \& Interactions}
We represent cells as spherical objects in 2D space.
They can be represented by their position $x_c$ and radius $r_c$ which is of course related to their (2D) volume
\begin{equation}
    V_c = \pi r_c^2
\end{equation}
As the coupled reactions from equations~\eqref{eq:spatial-full-system-simplified} change $V_c$, we also automatically adapt the cell radius accordingly.
Notice however, that how the radius is updated does not correspond to the cellular reactions defined previously but is a purely mechanical consideration.
The dimensionality of the problem does not find its way into the cellular reactions but only manifests itself in these mechanical considerations.
The underlying equation of motion is that of a standard damped particle
\begin{equation}
    m \ddot{\vec{x}} = -\eta\dot{\vec{x}} + \vec{F}
\end{equation}
Cells are interacting via a point-like interaction potential.
We can calculate the interaction strength via
\begin{alignat}{3}
    \vec{z} &= \vec{x_1} - \vec{x_2} \hspace{2em}&& r = ||\vec{z}||\\
    \sigma &= \frac{r}{r_1+r_2} && q=0.2\\
    \vec{F} &= \epsilon \frac{1+q}{q+\sigma} \theta(\sigma - 1) \frac{1}{r}\vec{z}
\end{alignat}
Cells are not sticky and thus only pushing on each other.
No attractive force is present.
This means that colonies are expanding as individual cells grow and push each other further apart.
%
% \begin{figure}
%     \centering
%     \includegraphics[width=0.6\textwidth]{Figures/interaciton_potential.png}
%     \caption{
%         Overall form of the interaction force between two cells.
%         A value $\sigma<1$ indicates that cells are overlapping which results in a positive force outwards.
%         Conversely, as long as cells are apart $\sigma>=1$, no force is acting.
%     }
%     \label{fig:spatial-interaction-potential}
% \end{figure}
%
%
% ................................................................................................
\subsubsection{Cell Cycle}
Typically, cell cycles consist of multiple phases.
Since our goal is to include only the necessary details of the underlying biological problem, we drastically simplify the description of the cell cycle.
Cell growth is already being described the system of coupled \acp{ode} given in equations~\eqref{eq:spatial-full-system-simplified}.
Thus, we introduce an additional rule that splits apart the cell whenever a certain volume threshold is reached.
The volume of the cell is divided equally between the new and old cell and both are assigned new random positions close to each other such that their radii do touch but not overlap.
\begin{figure}
    \begin{center}
    \includegraphics[width=0.49\columnwidth]{Figures/snapshot-2-cells-before-division.png}%
    \hspace{0.01\columnwidth}%
    \includegraphics[width=0.49\columnwidth]{Figures/snapshot-2-cells-after-division.png}
    \caption{
        (A) A cell right before and (B) after a division event.
        The volume is equally distributed between the two cells and the new positions are picked
        random but at opposing sides on a circle around the original center of the single cell.
    }
    \end{center}
\end{figure}
In addition to modeling division events, the cells are subject to the lag-phase which delays their ability to proliferate.
We model this by assigning a transition rate to each cell.
For each time point of the simulation, we generate a boolean value from a random number generator with probability
\begin{equation}
    P(\text{transition}) = \Delta t \lambda \frac{[R]}{[R_0]}.
\end{equation}
If this boolean is true, we move the cell from the inactive state to the active state where it starts to consume nutrients and grow.
It is clear that this is exactly identical to the \ac{ode} model.
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Comparing \acs{abm} and \acs{ode} Models}
The \ac{abm} model is more fundamental than the \ac{ode} model in the sense that it incorporates more biological aspects and maps them closer to realistic biological mechanisms.
We are thus able to derive the \ac{ode} pool model from its \ac{abm} version in the homogenous case (see supplementary material~\ref{subsec:supplement-deriving-pool-odes}).
By doing so, we find a mapping between parameters from the \ac{abm} to parameters used in the \ac{ode} description.
The derived equations read
\begin{align}
    \dot{R} &= -\frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V\\
    \dot{I} &= b V\\
    \dot{V} &= \frac{u}{1+\nu_I V_D^{-1} I} \frac{1}{V_D}\sigma R V.
\end{align}
and the \ac{ode} equations are
\begin{align}
    \dot{L_A} &= -\lambda_A R L_A\\
    \dot{G_A} &= \lambda_A R L_A + \alpha_A R G_A\\
    \dot{L_B} &= -\lambda_B R L_B\\
    \dot{G_B} &= \lambda_B R L_B + \frac{\alpha_B}{1 + \mu_I I} R G_B\\
    \dot{R} &= -\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{(1 + \mu_I I) N_t} R G_B \\
    \dot{I} &= \kappa G_A
\end{align}
We compare the most general version and thus omit species $A$ for the moment.
\begin{table*}
    \centering
    \renewcommand{\arraystretch}{1.5}
    \begin{tabularx}{0.8\textwidth}{@{}l *3{>{\centering\arraybackslash}X}@{}}
        & \textbf{\acs{ode}}
        & \textbf{\acs{abm}}
        & \textbf{Mappings}\\

        \toprule

        \multirow{2}{*}{Resource}
        & \multirow{2}{*}{$\displaystyle \dot{R} = - \frac{\alpha}{(1 + \mu_I I) N_t} R G$}
        & \multirow{2}{*}{$\displaystyle \dot{R} = -\frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V$}
        & $\displaystyle N_t = [R_0] V_D \sigma$\\

        &&& $\displaystyle \mu_I = \frac{\nu_I}{V_D}$\\

        Volume
        & $\displaystyle \dot{G} = \frac{\alpha}{1 + \mu_I I} R G$
        & $\displaystyle \dot{V} = \frac{u}{1+\nu_I V_D^{-1} I} \frac{1}{V_D}\sigma R V$
        & $\displaystyle \alpha = u[R_0]\sigma$\\

        Inhibitor
        & $\displaystyle \dot{I} = \kappa G_A$
        & $\displaystyle \dot{I} = b V$
        & $\displaystyle \kappa = b$\\
        \bottomrule
    \end{tabularx}
    \caption{Mappings between \ac{ode} and \ac{abm} parameters.}
\end{table*}
We can use these mappings to numerically compare both approaches.
Table~\ref{tab:spatial_limit_init} shows a comparison of parameters of two simulations for the \ac{ode} and \ac{abm} model which have been mapped onto each other.
Figure~\ref{fig:spatial-snapshots} shows individual snapshots of different 
%
\begin{figure}
    \includegraphics[width=0.49\columnwidth]{Figures/abm-homogenous/snapshot_00006000.png}\hspace{0.01\columnwidth}%
    \includegraphics[width=0.49\columnwidth]{Figures/abm-homogenous/snapshot_00012000.png}\vspace{0.01\columnwidth}\\
    \includegraphics[width=0.49\columnwidth]{Figures/abm-homogenous/snapshot_00018000.png}\hspace{0.01\columnwidth}%
    \includegraphics[width=0.49\columnwidth]{Figures/abm-homogenous/snapshot_00024000.png}%
    \caption{
        Simulation snapshots at $1.04$\unit{\day},$2.08$\unit{\day},$3.13$\unit{\day} and $4.17$\unit{\day} corresponding to the parameters given in table~table~\ref{tab:spatial_limit_init}.
        Species $B$ is colored blue while species $A$ is colored in red.
        The background shows the resource concentration $[R]$.
        Darker colors indicate that less nutrients are present.
    }
    \label{fig:spatial-snapshots}
\end{figure}
%
%
We see that in the highly-diffusive homogenous case of subfigure (a), the pool model is a good fit to the \ac{abm} model.
Heterogeneity and slower diffusion cause larger differences between the two models as can be seen in subfigures (b) and (c).
The combination of the two effects shows the largest difference between model predictions.
In subfigure (c), due to the heterogenous cell distribution (see also~\ref{fig:supplement-spatial-initial-config-cells}), overall inhibition of species $B$ is significantly reduced.
Subfigure (d) shows how a randomized homogenous initial distribution can lead to even more notable effects of inhibition.
We furthermore note that the total bacteria volume of the \ac{abm} model is always smaller than the \ac{ode} model.
This is expected, since processes such as diffusion and cells growing outward introduce another time component which is simply assumed to be instantanous in the \ac{ode} model.
The chosen diffusion constants $D_I=D_R=30.0$\unit{\micro\metre\squared\per\minute} are in the order of small molecules such as $NaCl$ while the lower values $D_I=D_R=5.0$\unit{\micro\metre\squared\per\minute} are around values of Glucose in water.
\begin{figure}
    \centering
    \begin{subfigure}[c]{0.5\columnwidth}
        \includegraphics[width=\columnwidth]{Figures/abm-homogenous/abm_ode_comparison.png}
        \subcaption{Homogenous distribution, $D=30.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}%
    \begin{subfigure}[c]{0.5\columnwidth}
        \includegraphics[width=\columnwidth]{Figures/abm-homogenous-low-diffusion/abm_ode_comparison.png}
        \subcaption{Homogenous distribution, $D=5.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}\\
    \begin{subfigure}[c]{0.5\columnwidth}
        \includegraphics[width=\columnwidth]{Figures/abm-inhomogenous/abm_ode_comparison.png}%
        \subcaption{Heterogeneous distribution, $D=30.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}%
    \begin{subfigure}[c]{0.5\columnwidth}
        \includegraphics[width=\columnwidth]{Figures/abm-random/abm_ode_comparison.png}%
        \subcaption{Randomized distribution, $D=5.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}
    \caption{
        Comparison of the \ac{ode} and \ac{abm} model for different parameter configurations.
        Snapshots of the spatial configurations of cells are included in the appendix~\ref{fig:supplement-spatial-initial-config-cells}.
        The diffusion constants are chosen such that $D_R=D_I=D$.
    }
    \label{fig:spatial-growth-curve}
\end{figure}
%
%
% ================================================================================================
\section{Discussion}

How well does the pool model do to encapsulate all the different growth models
\begin{itemize}
    \item All 6 models can be represented by pool model
    \item Baranyi-Roberts results can be reproduced, but not identical; requires some work
    \item what are some general limitations within this framework? how can they be circumvented?
    \item why is the pool model "better" than the other \ac{ode} models? (why did we do this?);
        generalization of many models; better explainability;
    \item link to supplement; we can generalize this and prove theorems for the abstract,
        generalized model; this allows us to discuss many cases at once
\end{itemize}

How justified was the construction of the \ac{abm}?
\begin{itemize}
    \item Discuss simplified physical interaction model
    \item Heterogeneity plays a role; as we have shown
    \item why did we not simply use a PDE?; because we have to think in terms of cellular processes
        and this ensures that our model is actually realistic
\end{itemize}

How can we cover more effects of bacteria?
\begin{itemize}
    \item More states for metabolic processes
    \item More states for more species
    \item Other cooperiation/inhibition effects
    \item external influences
\end{itemize}

What are limitations of our approach?
\begin{itemize}
    \item spatial stuff; heterogeneity (because \ac{ode})
    \item larger models require more parameters, harder to estimate
    \item only simple models have analytical solutions
\end{itemize}

% ------------------------------------------------------------------------------------------------
\section{Conclusion}
\begin{itemize}
    \item New theoretical framework
    \item consistent with \ac{ode} models (Baranyi-Roberts)
    \item Better explainability
    \item Include effects such as inhibition, toxins in understandable manner
    \item showed spatial limitations of \ac{ode} model
    \item \ac{abm} model allowed modeling of cellular processes; ensure that pool model is derivable
        from the individual behvaiour of cellular processes
    \item outlook: generalized mathematical formulation; extend inhibition model (because it is a
        dimensional reduction of existing models)
    \item Provide code in a pypi package as well (TODO)
\end{itemize}

\onecolumn

\newpage
\section*{Data Availability \& Reproducibility}

The data and simulation code which are presented in this study are publicly available in
\url{https://github.com/polinagaindrik/pool_model_paper}.

\printbibliography

% ================================================================================================
\newpage
\beginsupplement
\section{Supplementary Material}
% ================================================================================================
\subsection{General Mathematical Formalism}  \label{ssec:supplement1}
Consider the system consisting of $N$ species in $M$ different states.
In the general representation, all discussed in the main part systems  can be described with the Ordinary Differential Equations (\acp{ode}) using a non-linear function $\mbF$:
\begin{equation}
   \dot{\mbx} = \mbF(\mbx, \mbm, \mbp, \mbu),
\label{eq:model_ODE_x}
\end{equation}
where  $\mbx \in \mathbb{R}^{N}  \otimes \mathbb{R}^{M}$ is a vector describing bacterial count.
In matrix form it can be written as
\begin{equation}
    \mbx = \begin{pmatrix}
        x^1_1  & \dots & x^1_N  \\
        \vdots &       & \vdots \\
        x^M_1  & \dots & x^M_N  \\
            \end{pmatrix}
    \label{eq:model_bact}
\end{equation}
with $x_{j}^{i}$ describing the bacterial counts for species $i$ in state $j$.
$\mbu$ is the vector of different inputs of the model, e.g., temperature, pressure etc.
The vector $\mbm \in \mathbb{R}^{K}$ includes all variables corresponding to the microenvironment of the system which can be concentration of the inhibitors or activators:
\begin{equation}
    \dot{\mbm} = \mbH (\mbx, \mbm, \mbp, \mbu).
    \label{eq:model_microenv}
\end{equation}
Here function $\mbH$ is in general case also non-linear.\\
%
The advantage of the following formulation is its generality.
Depending on the knowledge about the system as well as the needed level of accuracy one can define an arbitrary number of states $M$ and control the complexity of the model.
Moreover, opposite to the already known Predictive Microbiology like Baranyi and Roberts one can easily change.\\
%
To calculate the bacterial count for one species $i$, we simply need to sum the tensor/vector $\mbx$ over all possible states:
\begin{equation}
    N_i = \sum_k \mbx_i^k.
\label{eq:N_pro_species}
\end{equation}
While the total bacterial count of all species of our system is
\begin{equation}
    N = \sum_i \sum_k \mbx_i^k.
\label{eq:N_total}
\end{equation}
%
%
%The three pool system described with equations~\ref{eq:ode_3pools} will be written as
%\begin{equation}
%    \dot \mbx  = A \mbx, \hspace{3mm} A = \begin{pmatrix}
%        -(\mu + \lambda) & 0             & 0      \\
%        \lambda          & \alpha - \mu' & 0      \\
%        \mu              & \mu'          & -\beta 
%    \end{pmatrix}.
%\label{eq:x_repres_3pools}
%\end{equation}
%
Using such matrix representation of the bacterial system, we can rewrite, for example, the three-pool model with limiting resource described with equations~\ref{eq:ode_3pool_resource} in a more compact form.
The concentrations of the pools are then the components of our variable vector $\mbx (t) = (L(t), G(t), D(t))^T$.
%
\begin{align}
    \dot \mbx  &= \mbF(\mbx, \mbm, \mbp, \mbu) = \begin{pmatrix}
        -(\mu + \lambda) & 0               & 0      \\
        \lambda          & \alpha R - \mu' & 0      \\
        \mu              & \mu'            & -\beta 
    \end{pmatrix} \mbx\\
    \dot \mbm &= \mbH (\mbx, \mbm, \mbp, \mbu) = -\frac{\alpha}{N_t} R G.
\end{align}
Separating linear and non-linear parts in two terms yields
\begin{equation}
    \dot \mbx = \begin{pmatrix}
        -(\mu + \lambda) & 0       & 0 \\
         \lambda         & \alpha  & 0 \\
         \mu &  \mu'& - \beta 
    \end{pmatrix} 
    \mbx + \frac{\alpha}{N_t} x^2 \sum_i^3 x^i \begin{pmatrix} 0 \\ 1 \\ 0  \end{pmatrix}.
\label{eq:x_repres_3pools_resource2}
\end{equation}
Here the first term is the linear part and the second term define the quadratic behavior of the system.
%
\paragraph{Generalized Lotka-Volterra}

Likewise, the multispecies system with resource competition present described by equations~\ref{eq:model_2sp_resource_comp} can be represented in the same way by defining
\begin{equation}
    \mbm = R, \hspace{1cm}
    \mbx = \begin{pmatrix}
        L_A & L_B \\
        G_A & G_B 
    \end{pmatrix}.
\label{eq:gLV_x_m}
\end{equation}

The simpler version of one state per species (Eq.~\ref{eq:LV_simple}) can be simply written in a general mathematical form as generalized Lotka-Volterra:
\begin{equation}
    \frac{d x_i}{d t} = x_i f_i (\mbx), \hspace{3mm} \mathbf{f} = \mathbf{r} + A \mbx.
\label{eq:gLV}
\end{equation}


\subsection{ABM Details} \label{ssec:supplement2}
\subsubsection*{Deriving the Pool Model}
\label{subsec:supplement-deriving-pool-odes}
Since the \ac{abm} can be considered more fundamental than the \ac{ode} model, we should be able to derive the latter one from the former.
This can be achieved by starting with equations~\eqref{eq:spatial-full-system-simplified}.
We integrate the spatial equations over the total domain starting with the resource concentration $[R]$.
\begin{alignat}{7}
    \int\limits_D\dot{[R]}\text{dVol} 
    &= \int\limits_D\bigg[&&\sum\limits_c &&-\frac{u}{1+\nu_I [I]} \frac{V_c}{V_D} [R] \delta(x-x_c) &&+ D_R\Delta [R]\bigg]\text{dVol}\\
    &= \int\limits_D\bigg[&&\sum\limits_c &&-\frac{u}{1+\nu_I [I]} \frac{V_c}{V_D} [R] \delta(x-x_c)\bigg]\text{dVol} &&+ \int\limits_D D_R\Delta [R]\text{dVol}\\
    &= \int\limits_D\bigg[&&\sum\limits_c &&-\frac{u}{1+\nu_I [I]} \frac{V_c}{V_D} [R] \delta(x-x_c)\bigg]\text{dVol} &&+ \int\limits_{\partial D} D_R\vec{\nabla} [R]\vec{\text{dA}}\\
    &= &&\sum\limits_c &&-\frac{u}{1+\nu_I [I]} V_c \int\limits_D \frac{1}{V_D} [R] \delta(x-x_c)\text{dVol}\\
    &= &&\sum\limits_c &&-\frac{u}{1+\nu_I [I]} V_c [R]_{x=x_c}\\
    &= &&\sum\limits_c &&-\frac{u}{1+\nu_I [I]} V_c [R]\\
    &= &&&&-\frac{u}{1+\nu_I [I]V_D V_D^{-1}} V [R]\frac{V_D}{V_D}\\
    \dot{R} &= &&&&-\frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V\\
\end{alignat}
We can split up the integral into two parts.
The second part can then be transformed into an integral over the flux of the resource along the boundary.
Since we chose Neumann boundary conditions, this contribution vanishes identically.
Afterwards, the first part can be evaluated by using the delta-distribution.
Note that we additionally need to consume a $V_D^{-1}$ term since our volume is finite.
We evaluate the resource concentration $[R]$ at the position of the individual cells and sum over all contributions.
Now we assume homogeneity in the sense, that we can reduce $[R]$ to only a single variable instead of considering its spatial distribution.
This means that every contribution does not depend on the position at which it was evaluated, ie. $[R]_{x=x_c}=[R]$.
Finally, we can perform the sum to gain the total volume of all cells $V=\sum\limits_c V_c$.
\begin{alignat}{7}
    \int\limits_D\dot{[I]} &= \int\limits_D \bigg[&&\sum\limits_c b\frac{V_c}{V_D}&&\delta(x-x_c) &&+ D_I\Delta [I]\bigg]\text{dVol}\\
    &= \int\limits_D&&\sum\limits_c b\frac{V_c}{V_D}&&\delta(x-x_c) \text{dVol} &&+ \int\limits_D D_I\Delta [I]\text{dVol}\\
    &= \int\limits_D&&\sum\limits_c b\frac{V_c}{V_D}&&\delta(x-x_c) \text{dVol} &&+ \int\limits_{\partial D} D_I\nabla [I]\vec{\text{dA}}\\
    &= &&\sum\limits_c b V_c\int\limits_D \frac{1}{V_D} &&\delta(x-x_c) \text{dVol}\\
    \dot{I} &= b V
\end{alignat}
Following the same principle as before, we can integrate the inhibitor equation.
For the equation describing the growth of the cell volume, we simply add up all cells to obtain the total volume of all cells.
\begin{alignat}{3}
    \sum\limits_c\dot{V_c}
    &= \sum\limits_c&&\frac{u}{1+\nu_I [I]} \sigma [R]V_c\\
    &= &&\frac{u}{1+\nu_I [I]} \sigma [R]V\\
    \dot{V} &= &&\frac{u}{1+\nu_I V_D^{-1} I} \frac{1}{V_D}\sigma R V
\end{alignat}
We can thus summarize the equations obtained
\begin{align}
    \dot{R} &= -\frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V\\
    \dot{I} &= b V\\
    \dot{V} &= \frac{u}{1+\nu_I V_D^{-1} I} \frac{1}{V_D}\sigma R V.
\end{align}
It is clear that the original model is recovered this way.
To obtain the full equations, we additionally need to consider transition between states of lag-phase and growth phase not depicted here.
The preceding equations all assume that cells are in growth phase.
It is however straiforward, to setup these equations.
%
%
% ................................................................................................
\subsubsection{Code Samples}
The code samples show sample code from the simulation that calculate cellular interactions and cell cycle.
Since backend of the simulation is written in Rust, so is the implementation of cellular aspects.
Additionally bindings are generated such that the full power of the simulation tool can be accessed.
This significantly improves handling and development time of new models.
\begin{listing}
\begin{lstlisting}[language=rust]
let z = ext_pos - own_pos;
let r = z.norm();
let sigma = r / (self.cell_radius + ext_radius);
if sigma < 1.0 {
    let q = 0.2;
    let dir = z.normalize();
    let modifier = (1.0 + q) / (q + sigma);
    return self.potential_strength * dir * modifier;
} else {
    return Vector2::zero();
}
\end{lstlisting}
\caption{
    Sample code for calculation of interactions between two bacteria cells.
}
\label{listing:spatial-interaction-strength}
\end{listing}
%
%
\begin{listing}
\begin{lstlisting}[language=rust]
// Clone existing cell
let mut c2 = c1.clone();

let r = c1.interaction.cell_radius;

// Make both cells smaller
c1.interaction.cell_radius /= std::f64::consts::SQRT_2;
c2.interaction.cell_radius /= std::f64::consts::SQRT_2;

// Generate cellular splitting direction randomly
let angle_1 = rng.gen_range(0.0..2.0 * std::f64::consts::PI);
let dir_vec = nalgebra::Rotation2::new(angle_1) * Vector2::from([1.0, 0.0]);

// Define new positions for cells
// It is randomly chosen if the old cell is left or right
let offset = dir_vec * r / std::f64::consts::SQRT_2;
let old_pos = c1.pos();

// Increase the remaining food in the bacteria by the fraction of volumes
// from before and after the division. This exactly coincides with factor 2
// since we half the volume of the cell
c1.cellular_reactions.intracellular_concentrations[0] *= 2.0;
c2.cellular_reactions.intracellular_concentrations[0] *= 2.0;

c1.set_pos(&(old_pos + offset));
c2.set_pos(&(old_pos - offset));
\end{lstlisting}
\caption{
    Sample code for dividing one bacteria cell into two.
}
\end{listing}
%
%
% ................................................................................................
\subsubsection{Full List of Parameters}
\label{sec:supplement-full-parameters}

\begin{table}
    \centering
    \begin{tabularx}{\textwidth}{@{}l *2{>{\centering\arraybackslash}X} l *2{>{\centering\arraybackslash}X}@{}}
        \multicolumn{3}{c}{\textbf{\acs{ode}}} & \multicolumn{3}{c}{\textbf{\acs{abm}}}\\
        \toprule
        \multicolumn{6}{c}{\textbf{Cellular Reactions}}\\
        \toprule
        Production $G_A$            & $\alpha_A$    & $0.002$\unit{\per\minute}
        & Uptake                    & $u_A$         & $0.01$\unit{\per\minute}\\
        Production $G_B$            & $\alpha_B$    & $0.002$\unit{\per\minute}
        & Uptake                    & $u_B$         & $0.01$\unit{\per\minute}\\
        Production $I$              & $\kappa$      & $0.1$\unit{\per\minute}
        & Production $I$            & $b$           & $0.1$\unit{\per\minute}\\
        Inhibition                  & $\mu_I$       & $1.0\cdot 10^{-7}$\unit{\per\micro\metre\cubed}
        &                           & $\nu_I$       & $0.1$\unit{\micro\metre\cubed\per\mol}\\
        &&
        & Conversion                & $\sigma$      & $0.1$\unit{\micro\metre\cubed\per\mol}\\
        Total volume                & $N_t$& $2.0\cdot 10^5$\unit{\micro\metre\cubed}
        &&&\\
        &&
        & Diffusion $R$             & $D_R$         & $30.0$\unit{\micro\metre\squared\per\minute}\\
        &&
        & Diffusion $I$             & $D_I$         & $30.0$\unit{\micro\metre\squared\per\minute}\\
        % Note: this diffusion roughly corresponds to the one of ethanol in water
        % (see https://de.wikipedia.org/wiki/Diffusionskoeffizient#In_Fl%C3%BCssigkeiten)
        % 15.0 µm^2/min = 0.9 m^2/s
        % Ethanol: 0.84 m^2/s
        \toprule
        \multicolumn{6}{c}{\textbf{Physical Mechanics \& Interactions}}\\
        \toprule
        &&& Damping                 & $\eta$        & $0.5$\unit{\per\minute}\\
        &&& Mass                    & $m$           & $7.70$\unit{\pico\gram}\\
        &&& Potential               & $\epsilon$    & $0.5$\unit{\pico\gram\micro\metre\per\minute\squared}\\
        \toprule
        \multicolumn{6}{c}{\textbf{Cell Cycle}}\\
        \toprule
        Rate $L_A\rightarrow G_A$   & $\lambda_A$   & $0.0025$\unit{\per\minute}
        &                           & $\lambda_A$   & $0.0025$\unit{\per\minute}\\
        Rate $L_B\rightarrow G_B$   & $\lambda_B$   & $0.0005$\unit{\per\minute}
        &                           & $\lambda_B$   & $0.0005$\unit{\per\minute}\\
        &&& Division Vol.           & $V_t$         & $14.14$\unit{\micro\metre\cubed}\\
        \toprule
        \multicolumn{6}{c}{\textbf{Initial Values}}\\
        \toprule
        Volume of $L_A$             & $L_A$         & $127.2$\unit{\micro\metre\cubed}
        & Cell Number               & $C_A$         & 18\\
        Volume of $G_A$             & $G_A$         & $0.0$\unit{\micro\metre\cubed}
        & Cell Volume               & $V_{c,A}$     & $7.07$\unit{\micro\metre\cubed}\\
        Volume of $L_B$             & $L_B$         & $127.23$\unit{\micro\metre\cubed}
        & Cell Number               & $C_B$         & 18\\
        Volume of $G_B$             & $G_B$         & $0.0$\unit{\micro\metre\cubed}
        & Cell Volume               & $V_{c,B}$     & $7.07$\unit{\micro\metre\cubed}\\
        Relative Nutrients          & $R$         & $1.0$
        & Nutrients                 & $[R_0]$       & $1.0$\unit{\mol\per\micro\metre\cubed}\\
        Inhibitor                   & $I_0$         & $0.0$\unit{\mol}
        & Inhibitor                 & $[I_0]$       & $0.0$\unit{\mol\per\micro\metre\cubed}\\
        \bottomrule
    \end{tabularx}
    \caption{
        Comparison of the parameters for the \ac{ode} model and \ac{abm}.
        % To view, how these parameters are stored inside a cell-agent, we refer to the supplementary
        % material~\ref{sec:supplement-full-parameters}.
    }
    \label{tab:spatial_limit_init}
\end{table}
%
% ................................................................................................
\subsubsection{Homogenous and Inhomogenous Initial Cell Distributions}
To clarify what we mean by homogenous and inhomogenous initial configurations, we include snapshots of the simulation.
At this point, cells have already grown such that the sizes of cell-clumps are better visible by the naked eye.
The snapshots were taken after $2.08$\unit{\day}.
\begin{figure}
    \centering
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures/abm-homogenous/snapshot_00012000.png}
        \subcaption{Homogenous initial cell distribution, $D_R=D_I=30.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}%
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures/abm-homogenous-low-diffusion/snapshot_00012000.png}
        \subcaption{Homogenous initial cell distribution, $D_R=D_I=5.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}\\
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures/abm-inhomogenous/snapshot_00012000.png}%
        \subcaption{Heterogeneous initial cell distribution, $D_R=D_I=30.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}%
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures/abm-random/snapshot_00012000.png}%
        \subcaption{Random initial cell distribution, $D_R=D_I=5.0$\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}
    \caption{
        Spatial configuration of cells corresponding to the pictures shown in figure~\ref{fig:spatial-growth-curve}.
    }
    \label{fig:supplement-spatial-initial-config-cells}
\end{figure}
%
%
\end{document}
