% \documentclass[10pt,A4paper]{article}
\documentclass[10pt,twocolumn,5p,authoryear]{elsarticle}
\usepackage[utf8]{inputenc}

\setlength{\tabcolsep}{1.3pt}
\setlength{\columnsep}{0.4cm}
\usepackage{acronym}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{array}
\usepackage{bm}
\usepackage{booktabs}
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off, font=small]{caption}
\usepackage{diagbox}
\usepackage{float}
\usepackage[T1]{fontenc}
% \usepackage[a4paper,inner=2cm,outer=2cm,top=2.5cm,bottom=2.5cm]{geometry}
\usepackage{graphicx}% Include figure files
\usepackage{here}
\usepackage{hyperref}
\usepackage{lipsum}
\usepackage[switch]{lineno}
\usepackage{mathtools}
\usepackage{mathrsfs}
\usepackage{multirow}
\usepackage{newfloat}
\usepackage{onimage}
\usepackage{siunitx}
\usepackage{subcaption}
\usepackage{tabularx}
\usepackage{tikz}
%\usepackage[dvipsnames]{xcolor}
\usepackage{xpatch}

% Ensure that subcaptions have captial letter
\renewcommand{\thesubfigure}{\Alph{subfigure}}

% Must be loaded after tikz
\usepackage{ctable}

% Define new column type with multirow
\newcolumntype{M}[1]{>{\centering\arraybackslash}m{#1}}

\bibliographystyle{elsarticle-harv}

\restylefloat{table}

% Remove indent across the whole document
\setlength\parindent{0pt}

% Define equation numbering
\numberwithin{equation}{section}

\newcommand{\beginsupplement}{%
  \renewcommand{\thesection}{}
  \setcounter{section}{0}
  \renewcommand{\thesection}{S\arabic{section}}
  % \setcounter{table}{0}
  \renewcommand{\thetable}{S\arabic{table}}%
  % \setcounter{figure}{0}
  \renewcommand{\thefigure}{S\arabic{figure}}%
}
\newcommand{\mbx}{\mathbf{x}}
\newcommand{\mbm}{\mathbf{m}}
\newcommand{\mbp}{\mathbf{p}}
\newcommand{\mbu}{\mathbf{u}}
\newcommand{\mbF}{\mathbf{F}}
\newcommand{\mbH}{\mathbf{H}}
\newcommand{\cmt}[1]{\textcolor{red} {[#1]}}
\newcommand{\etal}{{\textit{et al. }}}

\newacro{ode}[ODE]{Ordinary Differential Equation}
\newacro{pde}[PDE]{Partial Differential Equation}
\newacro{glv}[gLV]{generalized Lotka-Volterra}
\newacro{abm}[ABM]{Agent-Based Model}


% Define new style for json listings
% Copied from https://www.latex4technics.com/?note=3QTU
\colorlet{punct}{red!60!black}
\definecolor{background}{HTML}{EEEEEE}
\definecolor{delim}{RGB}{20,105,176}
\colorlet{numb}{magenta!60!black}

\title{Pool Model - A unified Mathematical\\ Modeling Framework for Food Science}
\author[1]{Polina Gaindrik}
\author[1]{Jonas Pleyer\fnref{fn1}}
\author[1]{Christian Fleck\corref{cor1}}
\ead{christian.fleck@fdm.uni-freiburg.de}

\cortext[cor1]{Corresponding Author}
\fntext[fn1]{jonas.pleyer.org}
\affiliation[1]{
    organization={Freiburg Center for Data Analysis and Modeling},
    addressline={Ernst-Zermelo Str. 1},
    city={Freiburg im Breisgau},
    postcode={79104},
    country={Germany},
}

\date{
    \footnotesize
    \textsuperscript{\textbf{1}}
}

\begin{document}

\begin{abstract}
    Bacterial consortia are complex systems which are often comprised of multiple distinct species.
    Their dynamics on a population-based level often result from various intra- and inter-species interactions,
    including cooperation or competition.
    Various empirical and mechanistic models have been conceived and applied to problems related to food microbiology.
    In this work we present a generalized framework, which uses multiple distinct "pools" that can distinguish
    between microbial types in different metabolic states or describe the extracellular environment properties.
    It allows us to incorporate the underlying biological processes and construct \ac{ode} based models of
    different complexity levels.
    To illustrate the application scope of the model, multiple different examples of microbial systems
    are reported.
    Finally, to explore the limitations of the \ac{ode} formulation, we analyze the influence of spatial effects.
    For this, we construct an \ac{abm} for a two-species system, showcasing that the resulting pool model
    arises naturally from the individual behavior of many cellular agents.
    We further apply this \ac{abm} to study the role of properties such as diffusion and spatial heterogeneity
    on the dynamics of bacterial growth.
\end{abstract}

%%Graphical abstract
\begin{graphicalabstract}
%\includegraphics{grabs}
\end{graphicalabstract}

\begin{highlights}
    \item @Christian fill this
\end{highlights}

\begin{keyword}
    Bacterial Consortia
    \sep Competition and Cooperation
    \sep Mechanistic Modeling
    \sep Spatial Effects
    \sep Predictive Modeling
\end{keyword}

\maketitle
\linenumbers
%
% ================================================================================================
\section{Introduction}

% \paragraph{General Introduction to Bacterial Consortia}
% \begin{itemize}
%     \item Complex system; many players (cell-types, interactions)
%     \item processes on different scales (population, cellular, molecular, gene-level)
%     \item Food Safety, Production
%     \item Healthcare: Drug development
%     \item Environmental Sustainabiliy
% \end{itemize}
% Motivation: general, flexible approach for bacterial systems that allows easily modify the system depending on processes on different scales.
% Allows us to describe the processes within populations using different bacterial states as well as interactions between different populations using ODEs.

Bacterial consortia are highly complex systems which consist of multiple distinct cell types.
The underlying processes are interconnected with each other and occur on all scales, ranging from
population-based dynamics of colony growth over molecular processes such as metabolism to gene
expression.
Bacterial communities are ubiquitous and even able to survive under extreme conditions, making them
virtually omnipresent around any living system.
They play an especially important role within the context of food safety and production
being a key contributor to the spoilage of food products as well as fermentation processes.
Furthermore, effects such as biofilm formation can hinder drug treatment by shielding the bacterial
colony from antibiotic substances or immune response of the organism.
The natural bacterial processes are a crucial part of our planet's ecosystem, being responsible for
tasks such as nutrient cycling, decomposition of organic matter and oxygen production.
It is of great interest to understand the dynamics of these various processes.

% \paragraph{Experimental/Biological evidence/phenomena}
% \begin{itemize}
%     \item Lag Phase, Competition, Cooperation, Limiting Resource
%     \item Comprehensive reviews about competition/cooperation bacterial mechanisms: \citep{ghoul_ecology_2016, stubbendieck_bacterial_2016, hibbing_bacterial_2010, west_social_2007}
%     \item Spatial Effects: adhesion, migration, chemotaxis, diffusion
%     \item Quorum-sensing
%     \item Biofilm formation \citep{Jin2020}
% \end{itemize}

In order to achieve a comprehensive description of the target system, it is important to incorporate
all relevant biological phenomena arising in these bacterial systems.
During their reproductive process, bacteria undergo multiple phases~\citep{buchanan_when_1997}, starting in a lag phase which
is necessary to adapt to new environments and initially impedes the overall growth~\citep{rolfe_lag_2012, monod_growth_1949}.
Additionally, bacteria are able to cooperate, e.g., via complementary metabolic processes or inhibit
each other, e.g., by means of waste production as a metabolic byproduct~\citep{west_social_2007, rainey_evolution_2003, hibbing_bacterial_2010, stubbendieck_bacterial_2016}.
A limited supply of nutrients will also affect the overall reproductive capacity of the colony~\citep{monod_growth_1949}.
Bacteria have thus devised various survival strategies to ensure sufficient access to nutrients.
Effects on a single-cell level such as adhesion~\citep{htuson_bacteriasurface_2013}, migration~\cite{decoene_microscopic_2011} and chemotaxis can lead to complex
phenomena on a population-based level which can aid in the goal of survival.
Cell signaling, e.g., in the form of quorum-sensing can lead to collective emergent phenomena such as
biofilm formation which play an important role for many mature microbial communities~\citep{Jin2020,ng_bacterial_2009}.

% \paragraph{ODE models (predictive microbiology, existing, overview, most popular)}
% Existing models in Predictive Microbiology (Primary Models)~\citep{perez-rodriguez_predictive_2012}:
% \begin{itemize}
%     \item An overview of primary models used in food science can be found in~\citep{van_boekel_kinetic_2008}.
%     \item Baranyi-Roberts Model~\citep{baranyi_modeling_1993, baranyi_dynamic_1994}
%     \item modified Gompertz~\citep{Gompertz1825-wi,Gibson1987}
%     \item Buchanan et al.: three-phase linear model for lag phase, exponential growth phase, and stationary phase~\citep{buchanan_when_1997}
%     \item McKellar two-compartment model with bacteria in growing and non-growing state~\citep{mckellar_heterogeneous_1997}
%     \item Inactivation models
% \end{itemize}
% %
% Secondary models:
% \begin{itemize}
%     \item Polynomial models 
%     \item Square Root-Type Models, e.g., Ratkowski for temperature~\citep{ratkowsky_relationship_1982}
%     \item The Cardinal Parameter Model~\citep{zwietering_decision_1992}
% \end{itemize}

The first step in the construction of any mathematical model is to determine the required level of
detail with respect to the description scale and consequently which effects to include.
The bacterial growth curves let us describe the dynamics of the overall abundance of various
species.
The field of predictive microbiology has developed a variety of ODE-based models to capture
bacterial growth dynamics~\citep{perez-rodriguez_predictive_2012}.
Various \acp{ode} \textit{primary models} describe the bacterial growth curve under constant
environmental conditions.
Widely used examples include the modified Gompertz model~\citep{Gompertz1825-wi,Gibson1987}, Baranyi-Roberts
model~\citep{baranyi_modeling_1993, baranyi_dynamic_1994} and McKellar two-compartment model~\cite{buchanan_when_1997} which distinguishes between growing and non-growing bacteria.
To take into account changing environmental conditions, so-called \textit{secondary models}
formulate the model parameters as a function of external quantities such as temperature, pH or
humidity.
These can be simple polynomial models, square root-type models~\citep{ratkowsky_relationship_1982} or
more complex relations such as the cardinal parameter model~\citep{zwietering_decision_1992}.
The advantage of the \ac{ode} formulation lies in the predictive power and simplicity of usage.
However, due to their coarse-grained nature, they lack the ability to describe heterogeneity and
other spatial effects.

% \paragraph{Models for spatial effects; PDEs, \acp{abm}, others}
% \begin{itemize}
%     \item BSim \citep{Gorochowski2012,Dang2020}
%     \item Bacterial Branching patterns \citep{Kawasaki1997,Matsushita1998}
% \end{itemize}

% To incorporate spatial heterogeneity or other effects other modeling approaches have been devised.
% \acp{pde} follow a similar pattern as \acp{ode} but also distribute the described variables through
% space.
When dealing with spatial effects such as heterogeneity, we are required to use more extensive
modeling formulations such as \acp{pde} or \acp{abm}.
The former represent an intermediate step between the population-based \ac{ode} models and the
individually-based \acp{abm} and can be effective, especially when dealing with large crowds of
cells that can be approximated as a continuum.
On the other hand, \acp{abm} excel at describing the individual behavior of cells which can then
lead to more complex collective phenomena.
Tools such as BSim~\citep{Gorochowski2012,Dang2020}, AgentCell~\cite{emonet_agentcell_2005}, BacSim~\cite{kreft_bacsim_1998} or \texttt{cellular\_raza}~\cite{Pleyer2025} are
capable to describe large amounts of bacteria, performing various intracellular and extracellular
processes.
The latter has been used to adapt \ac{pde}-based models into an \ac{abm} formulation which describe
bacterial branching patterns~\citep{Kawasaki1997,Matsushita1998}.

% \paragraph{What is the problem/what is the question? How do we address this?}

In this work, we introduce a new \ac{ode}-based modeling framework which describes different
bacterial species, their metabolic states and external quantities such as nutrients as individual
pools.
The dynamics of bacterial growth can be understood in terms of transitions between the different
bacterial states, i.e., pools.
Our model also allows us to capture interactions between these bacterial states and the external
microenvironment.
Compared against the aforementioned methods, this approach turns out to be less complex with general
applicability, low number of parameters and greater explainability for the underlying biological
reality.
To illustrate the generality of the model, multiple examples are reported.
We analyze the spatial limitations of an interacting two-species model and show that it can be fully
understood on an individual-based level in terms of single-cell processes.
Furthermore, we assess how spatial heterogeneity~\citep{mckellar_heterogeneous_1997} can impact the shape of the growth curves.


% ================================================================================================
\section{One species modeling}
% ------------------------------------------------------------------------------------------------
\subsection{Three-pool model}
We begin our discussion by considering a simplistic system where only one species is present $N=1$.
Bacterial growth undergoes different consecutive phases~\citep{buchanan_when_1997}, starting with the so called lag-phase, during which no increase in the bacterial count occurs.
This phenomenon appears due to adaptations of the microbial community, for example, to the new environmental conditions.
This so-called preparation before the exponential growth includes synthesis of the needed compounds, initiation of the transcription~\citep{rolfe_lag_2012} and other physiological and regulatory mechanisms~\citep{monod_growth_1949}.
After this, the bacteria which have succeeded in this adaptation stage enter the exponential growth phase, where the process of cell division is predominant.
When the levels of nutrition are exhausted by the growing bacteria, the stationary state of the growth curve occurs.
In this state the cultures accumulate waste products of metabolic activity that lead to cell death in the bacterial colony.
At the same time dying cells release the nutrients for survivors to continue growing.
As a result the balance between the growing and dying processes leads to a constant bacterial count~\citep{navarro_llorens_stationary_2010}.
In the described simplified version of the real-life system three possible bacterial states are mentioned, corresponding to the underlying processes: adaptation, growth or death.
Motivated by this we divide a bacterial population into three pools (states) (Figure~\ref{fig:SchematicRep}), defining the fraction of the population in the lag phase $L(t)$, growing and dividing $G(t)$ and undergoing cell death $D(t)$.
\begin{figure}[t]
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures-TPM_fig.pdf}
    \caption{Schematic representation of exchange between three states or pools.}
    \label{fig:SchematicRep}
    \end{center}
\end{figure}
To determine the equations which describe the system's dynamics, we consider the flows between the different states.
The pools $G$ and $D$ are unidirectionally connected with the pool $L$.
For now, we assume no back flow from the growth state $G$ to the lag-state $L$.
The suggested model can be cast in the following reactions:
\begin{align}
    L &\stackrel{\lambda}{\longrightarrow} G\\
    L &\stackrel{\mu}{\longrightarrow} D\\
    G &\stackrel{\alpha}{\longrightarrow} 2G\\
    G &\stackrel{\mu'}{\longrightarrow} D
\end{align}
Then the \acp{ode} describing such a model can be written as
\begin{align}\begin{split}
    \dot{L} &= -(\mu + \lambda) L\\
    \dot{G} &= \lambda L + \alpha G - \mu' G\\
    \dot{D} &= \mu  L + \mu' G 
\end{split}\end{align}
with the initial conditions $L(0)=N_0$, $G(0)=0$, and $D(0)=0$, where $N_0$ is the initial bacterial population.
%
Based on experimental observations, it was shown that the bacterial growth is strongly influenced by the limiting resource.
According to the works of \citet{monod_growth_1949} in the system where all needed nutrients are in excess except for one, the total growth is linear to the concentration of this limiting nutrient resource.
The control of the population density with respect to the available nutrition by means of cell communication is called quorum sensing.
By releasing extracellular signalling molecules (autoinducers) it is possible for the bacteria to synchronize the gene expression of the whole group~\citep{ng_bacterial_2009}.
Instead of modeling complex signalling networks, we choose a simpler approach to describe the dependence of growth on the nutrient availability.
We include a resource pool which is consumed by the growing bacteria using the second order kinetic reaction:
\begin{equation}
    G + R  \stackrel{\alpha_0}{\longrightarrow} 2G
\end{equation}
allowing us to keep a minimal number of parameters.
Note that the pools $L$, $G$ and $D$ represent the number of bacteria (or number of bacteria per unit area/volume) while $R$ represents an abstract resource pool.
Let's say $G=N_B/V$ and $[G]=\unit{1/V}$, where $N_B$ is the number of bacteria and $V$ is the volume.
Then $R=N_R/V$ and $[R]=\unit{1/V}$, where $N_R$ is the number of resource molecules (or any appropriate unit, e.g., \unit{mol}).
Then $[\alpha_0]=\unit{V/s}$, $[N_t]=\unit{1/V}$, and $[\alpha]= \unit{1/s}$.
$N_t$ denotes the maximal size of the total bacterial population due to environmental limitations (Figure \ref{fig:3pool_resource_plots}).
For homogenous densities it is equivalent to consider the total amount, i.e., multiplying $L$ and $G$ by the volume.
In this case $[L]=[G]=[N_t]=1$.
For simplicity, we make a transition to the dimensionless constant $\alpha:=\alpha_0 N_t$ and dimensionless resource pool $\tilde{R}(t) = R/ N_t$ ($0 \leqslant \tilde{R} \leqslant 1$).
Here the full capacity of the resource pool $N_t$ was used as a scaling constant.
In this case the growth term can be simply written as $\alpha G \tilde{R}$ and the model is fully formulated in the following way:
\begin{align}
    \begin{split}
        \dot{L} &= -(\mu + \lambda) L\\
        \dot{G} &= \lambda L + \alpha \tilde{R} G-\mu' G\\
        \dot{D} &= \mu  L + \mu' G\\
        \dot{\tilde{R}} &= - \frac{\alpha}{N_t} \tilde{R} G
    \end{split}
\label{eq:ode_3pool_resource}
\end{align}
%
This can be simplified by exploiting the conservation of the resource pool
\begin{equation}
    \tilde{R} = 1 - \frac{L+G+D}{N_t}
\end{equation}
and, hence, the full system can be written as
\begin{align}
    \begin{split}
        \dot{L} &= -(\mu + \lambda) L\\
        \dot{G} &= \lambda L + \alpha G\left(1-\frac{L+G+D}{N_t}\right)-\mu' G\\
        \dot{D} &= \mu  L + \mu' G.
    \end{split}
\label{eq:ode_3pool_resource2}
\end{align}
%
%
\begin{figure}
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_3pools_resource.pdf}
    \caption{
        The time dependence of the bacterial count $N(t) = L(t) + G(t) $ is shown.
        All curves were calculated using the model (\ref{eq:ode_3pool_resource2}) with parameters $\alpha=2.5$, $\mu'=0.5$.
        The blue curve shows a significant dip in the behavior of the system where the death of the bacteria is present $\mu = 0.5$.
        Other curves show the case of no bacteria dying $\mu=0$ and represent different lengths of lag phases: short $\lambda=10^{-2}$ (dark green), intermediate $\lambda=10^{-3}$ (light green) and long one $\lambda=10^{-4}$ (yellow). Note that the steady state for $N(t)$ and $D(t)$ is for this model $\lim_{t\to\infty} N(t)=0$ and $\lim_{t\to\infty} D(t)=N_t$, because of the depletion of the resource pool. 
    }
    \label{fig:3pool_resource_plots}
    \end{center}
\end{figure}
The resulting growth curves for the three pool case are presented in Figure~\ref{fig:3pool_resource_plots}.
Here we can see that the value of the transition constant $\lambda$ determines how long the bacteria stays in its lag-phase.
The larger this rate, the shorter is the lag-time.
Moreover, by including the flow of living bacteria to the death pool, we can model a greater variety of effects by allowing a decrease in the bacterial count.
% TODO change the name "death pool"?
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Lag-time estimation}
Note that the previously introduced model could be further simplified to a two pool or state model by omitting the pool $D$ and removing cells directly from pools $L$ and $G$.
The relevant question here is whether cells in the pool $D$ still contribute to the limitation of the resources by, e.g., occupying space or consuming important nutrients.
If yes, the three pool model would be advantageous, if not the two pool model is sufficient with the benefit of having one parameter less.\\
%
% The dynamics of the three pool model can be captured by the coupled system of ordinary differential
The dynamics of the three pool model is captured by equations~\ref{eq:ode_3pool_resource2} which can be used to derive the two pool model simply by omitting $D$.
Moreover, for $\mu'=0$ the equations for $L$ and $D$ can be readily integrated resulting in:
\begin{align}
    \begin{split}
        L(t) &= N_0 e^{-(\mu+\lambda)t}\\
        D(t) &= N_0 \mu \frac{1-e^{-(\mu+\lambda)t}}{\mu+\lambda
       }
    \end{split}.
\end{align}
For constant parameters $\mu$ and $\lambda$ the integrals for $L$ and $D$ can be inserted into the differential equation for $G$.
However, in the case of time dependent parameters, e.g., for a dependence on a non-static, dynamic temperature, it appears to be not advantageous to use the analytical results for $L$ and $D$ and work instead with all three equations for $L$, $G$, and $D$.
It is interesting to see what type of equations result for the total bacterial population $\mathcal{N}=L+G+D$:
\begin{align}
    \begin{split}
        \dot{G} &= \lambda L + \alpha G\left(1-\frac{\mathcal{N}}{N_t}\right)\\
        \dot{\mathcal{N}} &= \alpha G\left(1-\frac{\mathcal{N}}{N_t}\right)
    \end{split}
\end{align}
with initial conditions $G(0)=0$ and $\mathcal{N}(0)=N_0$.
These equations have a clear interpretation, are derived from first principles, and are not identical to the popular Barayni-Roberts and modified-Gompertz models.
A notable difference is that the growth rate $\alpha$ is time independent.
For the functions $L$ and $D$ it holds: $\lim_{t\to\infty} L(t) = 0$ and  $\lim_{t\to\infty} D(t) =  N_0\frac{\mu}{\mu+\lambda} $.
It follows immediately that $\lim_{t\to\infty} G(t) = N_t-N_0\frac{\mu}{\mu+\lambda} $. The lag time $t_L$ can be approximated by the time at which half of the initial population is in the growth phase, i.e., $N_0=2G(t_L)$.
Assuming $L\ll N_t$, we can ignore the non-linear term in the equation for $G$, which gives rise to:
\begin{align}
    N_0&= 2\int_0^{t_L} \lambda L(t)e^{\alpha(t_L-t)}dt\\
    1  &= 2\lambda e^{\alpha t_L}\frac{1-e^{-(\mu+\lambda+\alpha)t_L}}{\mu+\lambda+\alpha}.
\end{align}
For $\alpha > \mu+\lambda$  we can approximate $t_L$ by:
\begin{align}
    t_L &\approx -\frac{1}{\alpha}\ln\left(\frac{2\lambda}{\mu+\lambda+\alpha}\right).
\end{align}
The lag time $t_L$ diverges very slowly, logarithmically with $\lambda\to 0$.
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Back flow from growth to lag pool}
Bacteria which have lived under rather unfavorable conditions have developed different surviving mechanisms, one of which is adapting their metabolism and transitioning into dormant state.
Dormancy in bacteria is a reversible state where cell continue living with low metabolic activity while the cell division stops~\citep{kaprelyants_dormancy_1993}.
This survival strategy can appear in response to different external stresses, e.g., nutrient deficiency, temperature stress or extreme fluctuations of environment parameters.
And after "difficult times" bacteria can continue cell division~\citep{kell_viability_1998}.
Multiple models which include transitions of bacteria to the dormant state were proposed by \citet{bar_modelling_2002} and \citet{jones_dormancy_2010}.
%
We explore now the idea that under certain conditions, a sudden change in the environment can lead to parts of the population $G$ entering a lag-phase.
Within the framework of our pool model this means, that there is a back-flow from $G$ to $L$.
\begin{align}
    L &\stackrel{\lambda}{\longrightarrow} G\\
    L &\stackrel{\mu}{\longrightarrow} D\\
    G &\stackrel{\alpha}{\longrightarrow} 2G\\
    G &\stackrel{\gamma(t)}{\longrightarrow} L
\end{align}
The rate $\gamma(t)$ depends on the rate of change of the environment.
Stronger and quicker changes will lead to a higher rate $\gamma$.
%
%
\paragraph{Maxwell type of stress - strain relation}
One of the reasons that bacteria enter the dormant state can be due to a shift in temperature~\citep{oliver_viable_1995}.
Moreover, a rapid shift in temperature is associated with the appearance of an additional lag phase~\citep{zwietering_modeling_1994}.
To model it we propose a kind of viscoelastic Maxwell type of stress-strain relation and write down the ordinary differential equation for $\gamma$ to be:
\begin{equation}
    \dot{\gamma} = \Gamma |\dot{T}|-\delta \gamma.
\end{equation}
We assumed that the direction of the temperature change does not matter, i.e., a change from $T=2$ to $T=12$ has the same effect as a change from $T=14$ to $T=4$.
$\Gamma$ is a scaling factor and $\delta$ is the relaxation time for the temperature disturbance.
Integration of this equation yields (with $\gamma(0)=0$):
\begin{equation}
    \gamma(t) = \Gamma \int_0^t |\dot{T}(t^{\prime})|e^{-\delta (t-t^{\prime})}dt^{\prime}.
\end{equation}
After some manipulation of this equation we find:
\begin{equation}
    \gamma(t) = \Gamma\delta\left |T[t]-T[0]e^{-\delta t}-\delta
        \int\limits_0^t T(t^{\prime})e^{-\delta (t-t^{\prime})}dt^{\prime}\right |.
\end{equation}
This representation of the rate $\gamma(t)$ has the advantage that it only involves $T(t)$ and does not require the calculation of the derivative of $T$.
If $T$ exhibits $n$ temperature jumps, one finds:
\begin{eqnarray}
    \gamma(t) &=& \Gamma\sum_{i=1}^n \left |\Delta T_i \right |\theta(t-t_i)e^{-\delta(t-t_i)}.
\label{eq:gamma_tempshift}
\end{eqnarray}
%
The ordinary differential equations for the pool model with back-flow then read:
\begin{align}
    \begin{split}
        \dot{L} &=\gamma(t) G - (\mu + \lambda) L\\
        \dot{G} &= \lambda L -\gamma(t) G + \alpha G\left(1-\frac{L+G+D}{N_t}\right)\\
        \dot{D} &= \mu  L
    \end{split}
    \label{eq:ode_tempshift_backlag}
\end{align}
\begin{figure}[t]
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_3pools_resource_tempshift1.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_3pools_resource_tempshift2.pdf}
    \caption{
        (A) The time dependence of the total bacterial count $N$ for the model including the back-flow to the lag phase due to temperature stress~(\ref{eq:ode_tempshift_backlag}).
        (B) The value of the corresponding back-flow rate $\gamma$ calculated according to (\ref{eq:gamma_tempshift}).
        The curves represent no temperature shift (green), one temperature jump $\Delta T = 10$ at time $t=4$ (orange)
        and the series of different temperature shifts of value 10, 5 and 15 at the time points 1, 3 and 7, respectively (rose).
        The parameters of the back-flow rate are $\Gamma=1$ and $\delta=5$.
        The other model parameters are $\alpha=2.5$, $\mu'=0.5$, $\beta=5$, $\lambda=10^{-2}$, $\mu = 0$ for each curve.
    }
    \label{fig:TempJump}
    \end{center}
\end{figure}
The effect of the temperature shift on the total bacterial count can be shown in Figure~\ref{fig:TempJump}.
After the rapid temperature increase, the growth slows down till the bacteria that transitioned back to lag-phase migrate back to the growth pool.
If the shift happened rather early in time, when most of the bacteria were in lag-phase, the effect is not noticeable.
%
% ================================================================================================
\section{Interaction between species}
As mentioned above, bacteria usually coexist in large interacting communities of multiple species whose growth is vastly determined by the developed survival mechanisms.
These mechanisms mostly aim to secure a constant availability of nutrition which leads to competitive strategies or, if it is advantageous, to cooperative ones~\citep{hibbing_bacterial_2010, stubbendieck_bacterial_2016}.
Further we show that the model can be readily extended to the systems with more than one bacterium and describe more complex relationships between species.
%
% ------------------------------------------------------------------------------------------------
\subsection{Resource competition}
One of the ways to describe competition between bacterial populations is as a function of the limiting resource ratio~\citep{tilman_resource_1977, smith_effects_2002,ghoul_ecology_2016}.
To this end we consider for simplicity only one common limiting resource pool.
The simplest pool model for bacterial species $A$ and $B$ encompassing lag-phases, growth-phases and a common resource pool which is given by (no back-flow to the lag-pool):
\begin{align}
    \begin{split}
        \dot{L}_A &= - \lambda_A R L_A\\
        \dot{L}_B &= - \lambda_B R L_B \\
        \dot{G}_A &= \lambda_A R L_A +\alpha_A R G_A\\
        \dot{G}_B &= \lambda_B R L_B +\alpha_B R G_B\\
        \dot{R} &=-\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{N_t} R G_B
    \end{split}
    \label{eq:model_2sp_resource_comp}
\end{align}
We again rescaled the resource pool with its total capacity $N_t$ and find
\begin{equation}
    R =1-\frac{L_A+G_A+L_B+G_B}{N_t}
    \label{eq:Resource}
\end{equation}
Inserting equation~\ref{eq:Resource} into system~\ref{eq:model_2sp_resource_comp} yields:
\begin{align}
    \dot{L}_A &= -\lambda_A  L_A + \frac{\lambda_A}{N_t}L_AN\\
    \dot{L}_B &= -\lambda_B L_B + \frac{\lambda_B}{N_t}L_B N\\
    \dot{G}_A &=  \lambda_A  L_A + \alpha_A G_A - \frac{\lambda_A L_A + \alpha_A G_A}{N_t} N\\
    \dot{G}_B &=  \lambda_B L_B - \alpha_B G_B \frac{\lambda_B L_B + \alpha_B G_B}{N_t}N
\end{align}
where we used the total bacterial count $N=(L_A+G_A+L_B+G_B)$.
These equations have the structure of a \ac{glv} model expanded with different internal states for one species.
As one can see in Figure~\ref{fig:2pool_resource_2sp} even without any direct interaction between species the bacterial growth was inhibited due to the limitation of the resource.
Here species $A$ has a longer lag-phase than $B$ but the exponential growth of $A$ is faster compared to $B$.
As a result, species $A$ was able to suppress $B$, consuming the larger amount of nutrients and reaching a larger bacterial count at the end.
\begin{figure}
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_2pools_resource_competition.pdf}
    \caption{
        The competition of the two bacterial species due to the common limiting resource calculated using equation~(\ref{eq:model_2sp_resource_comp}).
        The red curve describes the growth of the bacteria $A$ ($N_A = L_A+G_A$) with model parameters $\lambda_A=10^{-2}$, $\alpha_A=3.0$.
        The blue curve describes the growth of the bacteria $B$ ($N_B = L_B+G_B$) with model parameters $\lambda_B=5\cdot 10^{-2}$, $\alpha_B=2.5$.
        The purple line shows the total bacterial count of the system  $N(t)=N_A+N_B$.
        The gray line denotes the dynamic of the resource pool scaled with its capacity $N_t=10^4$.
    }
    \label{fig:2pool_resource_2sp}
    \end{center}
\end{figure}
%
In order to get the commonly used form of the \ac{glv}, we need to simply omit the bacterial states $L_A=L_B=0$:
\begin{align}
    \begin{split}
        \dot{G}_A &= \alpha_A G_A\left(1 - \frac{G_A}{N_t}\right) - \frac{\alpha_A}{N_t}G_AG_B\\
        \dot{G}_B &= \alpha_B G_B\left(1-\frac{G_B}{N_t}\right) -\frac{\alpha_B}{N_t}G_AG_B. 
    \label{eq:LV_simple}
    \end{split}
\end{align}
%
Equations (\ref{eq:LV_simple}) can be understood as competitive Lotka-Volterra equations that are widely used in ecological studies and show the principle of competitive exclusion.
That principle claims that two species which compete for the same resource cannot coexist~\citep{wangersky_lotka-volterra_1978}.
In previous studies it was shown that \ac{glv} equations with one bacterial state can be successfully applied to predict the microbial community dynamics and determine different types of the pairwise interspecies interactions~\citep{dedrick_when_2023, stein_ecological_2013, venturelli_deciphering_2018, hoffmann_power_2007}.
In food studies the \ac{glv} was applied by \citet{mounier_microbial_2008} to study interactions of six bacterial species influenced by yeast omissions in cheese.
\citet{cauchie_modeling_2020} applied the Lotka-Volterra model in their study of the three-species bacterial system of minced pork meat.\\
%
However, in contrast to the classical Lotka-Volterra approach, the introduced pool model allows for far more flexibility in tuning properties of the bacterial interactions within and between species.
For further comparison of the derived model with the matrix formulation of the \ac{glv} we refer the reader to the Appendix~\ref{ssec:supplement1}.
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Interspecies competition}
Mutual inhibition is often a result of active competition mechanisms which have been developed by bacteria to contend for a shared resource.
Some examples of such mechanisms are the production of antimicrobial compounds, inhibitors or toxins~\citep{wloch-salamon_effect_2008, chao_structured_1981}.
On the other hand, some bacteria can produce public goods, the compounds that benefit growth of other species.
In this way, they can suppress the non-preferred species by helping their competitors~\citep{hibbing_bacterial_2010}.
%
%
% ................................................................................................
\subsubsection{Toxin production}
One direct form of inhibition is the production of toxins.
To introduce active competitive mechanisms, we extend our model with one of the species, producing the toxic compounds $T$ during its growth:
\begin{equation}
    \dot{T} = \mathcal{F}(R,G).
\end{equation}
The function $\mathcal{F}$ denotes the model for the toxin production rate.
One example is the accumulation of bacterial waste which is responsible for the spoilage of food products and suppressing growth of other species.
Consider the system of two species competing for resource $R$ with the production of the toxin $T$ by species $B$.
We assume that the toxin causes the death of the species $A$ with rate $\mu$.
Then the full \ac{ode} system is
\begin{align}
    \begin{split}
        \dot{L}_A &= - \lambda_A R L_A\\
        \dot{L}_B &= - \lambda_B R L_B \\
        \dot{G}_A &= \lambda_A R L_A + \alpha_A R G_A - \frac{\mu}{\kappa N_t} T G_A\\
        \dot{G}_B &= \lambda_B R L_B + \alpha_B R G_B\\ 
        \dot{R} &= -\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{N_t} R G_B\\
        \dot{T}_{B}^{1,2} &= \mathcal{F}^{1,2} (R, G_B) \\
    \end{split}
    \label{eq:model_2sp_toxin}
\end{align}
%
where we consider two possible models of the toxin production
\begin{align}
    \mathcal{F}^1(R,G)&=\kappa G_B\\
    \mathcal{F}^2(R,G)&=\kappa\alpha_B R G_B.
\end{align}
with $\kappa$ as the toxin production rate.
The difference in the solutions of the given \acp{ode} between the two toxin production models is presented in Figure~\ref{fig:2pool_2sp_toxin}.
%
%
\begin{figure}
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_2pools_toxin1.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_2pools_toxin2.pdf}
    \caption{
        The two species system competing for the common nutrient $R$ with production of the toxin $T_B$ (brown line) calculated using equation~(\ref{eq:model_2sp_toxin}).
        The growing bacteria $B$ produces toxin $T_B$ (A) constantly $\dot{T}_B^1 = \mathcal{F}^1 = \kappa G_B$ or (B) only during growth process $\dot{T}_B^2 = \mathcal{F}^2 = \kappa \alpha_B R G_B$ with
        production rate $\kappa=0.2$.
        The toxin $T_B$ causes the cell death of bacteria $A$ with rate constant $\mu = 0.2$.
        The red curve describes the growth of the bacteria $A$ ($N_A = L_A+G_A$) with model parameters $\lambda_A=10^{-2}$, $\alpha_A=3.0$.
        The blue curve describes the growth of the bacteria $B$ ($N_B = L_B+G_B$) with model parameters $\lambda_A=5\cdot 10^{-2}$, $\alpha_A=2.5$.
        The gray line denotes the dynamic of the resource pool scaled with its capacity $N_t=10^4$.
    }
    \label{fig:2pool_2sp_toxin}
    \end{center}
\end{figure}
%
Model $\mathcal{F}^1$ results in toxin concentration $T_B^1(t)=\kappa\int_{t_0}^tG(t')dt'$.
The production of toxin only depends on the presence of bacteria $G_B$ and is independent of their growth dynamics.
% That means that as long as there are bacteria in the growth phase $G_B>0$, toxin is produced, no matter whether the bacterial count $N_B$ is still growing.
On the other hand, model $\mathcal{F}^2$ assumes that toxin is only produced during bacterial growth and division, hence depends on the growth dynamics of $B$.
This leads to a toxin accumulation in the steady state $T_B^{2s}=\lim_{t\to\infty}T_B^2(t)=\kappa(N_t-N_0)$.
To see this we note $\dot{T}_B^2=\kappa(\dot{L_B}+\dot{G_B})$, which yields: $T_B^2(t)=\kappa(L_B(t)+G_B(t)-N_{B0})$ where we used $T_B^2(0)=G_B(0)=0$, $L_B(0)=N_{B0}$.
%
% ................................................................................................
\subsubsection{Mutual inhibition}
Another special case of interspecies competition is when the produced toxin does not cause the death of the bacteria but rather inhibit their growth.
Such a situation with 2 bacteria species, a shared resource and produced inhibitor was modeled by \citet{lenski_coexistence_1986}.
The same effect can be modeled by utilizing our pool-based approach.
In our case, we would like to look at a rather simplified case reducing the number of variables in the system.
The easiest way to introduce inhibition of species $A$ by species $B$ is by assuming that the growth rate of species $A$ depends on the abundance of species $B$.
For the sake of simplicity we ignore the lag pool and describe the growth components via:
\begin{align}
    \dot{G}_A &= \mathcal{B}_A(G_B)G_A\left(1 - \frac{G_A+G_B}{N_t}\right)\\
    \dot{G}_B &= \mathcal{B}_B(G_A) G_B\left(1-\frac{G_A+G_B}{N_t}\right)
\end{align}
The functions $\mathcal{B}_{A/B}$ are a phenomenological choice to capture the inhibiting effects.
It is straightforward to make the mechanism of inhibition more explicit by, e.g., introducing a toxin pool.
One possible choice for $\mathcal{B}$ is:
\begin{equation}
    \mathcal{B}(G) = \frac{\alpha}{1+KI(G)}
\end{equation}
where $I$ is the concentration of the inhibiting metabolic byproduct of the bacterial growth.
This leads to
\begin{align}\begin{split}
    \dot{G}_A &= \frac{\alpha_A G_A}{1+K_BG_B}\left(1 - \frac{G_A+G_B}{N_t}\right)\\
    \dot{G}_B &= \frac{\alpha_B G_B}{1+K_AG_A}\left(1-\frac{G_A+G_B}{N_t}\right).
\end{split}\end{align}
% Note that if the toxin accumulates, i.e., $\dot I \sim G$,  one needs to replace $G$ by $\int G dt$.
One can distinguish two limiting cases: {\it i}) $K_{A/B}\ll N_t$ and {\it ii}) $K_{A/B}\gg N_t$.
In the first case the inhibition is of minor importance and could be ignored while in the latter case one can simplify the equations, given $N_A$ and $N_B$ are non-zero:
\begin{align}\begin{split}
    \dot{G}_A &=\tilde{\alpha}_A\frac{G_A}{G_B}\left(1 - G_A-G_B\right)\\
    \dot{G}_B &= \tilde{\alpha}_B\frac{G_B}{G_A}\left(1 - G_A-G_B\right).
\end{split}\end{align}
We introduced the rescaled growth rates $\tilde{\alpha}_A=\alpha_A/(N_tK_B)$ and $\tilde{\alpha}_B=\alpha_B/(N_tK_A)$.
Note that $[\tilde{\alpha}]=\mathrm{s}^{-1}$.
We rescale the time with $\tilde{\alpha}_A$ and define the dimensionless parameter
\begin{equation}
    \psi = \tilde{\alpha}_B/\tilde{\alpha}_A = \frac{K_B\alpha_B}{K_A\alpha_A}
\end{equation}
which results in
\begin{align}
    \begin{split}
        \dot{G}_A &=\frac{G_A}{G_B}\left(1 - G_A-G_B\right)\\
        \dot{G}_B &=\psi\frac{G_B}{G_A}\left(1-G_A-G_B\right). 
    \end{split}
\label{eq:2sp_1pool_inhibit_unitless}
\end{align}
We can further separate three cases
\begin{align}
    \psi&=1:\quad\text{A and B are equivalent}\\
    \psi&>1:\quad\text{B grows faster/inhibits A}\\
    \psi&<1:\quad\text{A grows faster/inhibits B}
\end{align}
which are displayed in Figure~\ref{fig:1pool_2sp_inhibit}.
The advantage of this system is its simplicity, allowing us to describe the effect of mutual inhibition between two bacterial species with only a single parameter.
%
\begin{figure}
    \begin{center}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_1pool_inhib1.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_1pool_inhib2.pdf}\\
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_1pool_inhib3.pdf}
    \caption{
        The two-species system mutually inhibiting each other is defined with equations~(\ref{eq:2sp_1pool_inhibit_unitless}).
        Here the red lines describes the evolution of the species $A$ and blue lines correspond to species $B$.
        The relation between inhibition strength of species $B$ and $A$ are defined by control parameter value (A) $\psi$ = 1, (B) $\psi$ = 0.5 or (C) $\psi$ = 2.
    }
    \label{fig:1pool_2sp_inhibit}
    \end{center}
\end{figure}
%
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Interspecies cooperation}
On the other hand, one can also observe some cooperative behaviors between species in cases when it facilitates the competition with others.
This phenomenon occurs due to complementary metabolic pathways, i.e., production of so-called public goods which are beneficial for the growth of other species.
It is most often observed between species which are close relatives or clonal populations.
This process is called kin selection~\citep{west_social_2007}.
Furthermore, neighbors can also benefit from a behavior which does not require any action on an expression level, so-called 'social cheaters'~\citep{rainey_evolution_2003}.
This mutual activation benefits all involved species and can be easily included in our mathematical framework.
For the sake of clarity we omit again the lag pool and write
\begin{align}
    \dot{G}_A &= \mathcal{B}_A(P_B,R)G_A\\
    \dot{G}_B &= \mathcal{B}_B(P_A,R)G_B\\
    \dot{R} &=-\mathcal{B}_A(P_B,R)G_A-\mathcal{B}_B(P_A,R)G_B\\
    \dot{P}_A &= \mathcal{F}_A(R,G_A)\\
    \dot{P}_B &= \mathcal{F}_B(R,G_B)
\end{align}
where $P_A$ and $P_B$ denote the substance produced by species A and B, respectively.
The functions $\mathcal{B}_A(P_B,R)$ and $\mathcal{B}_B(P_A,R)$ capture the resource dependent growth.
Setting $\mathcal{B}_A(P_B,R)=\alpha_A P_BR$, $\mathcal{B}_B(P_A,R)=\alpha_B P_A R$, $\mathcal{F}_A(R,G_A)=\kappa_A RG_A$, $\mathcal{F}_B(R,G_B)=\kappa_B RG_B$, rescaling $G_A$, $G_B$, and $P_B$ with $N_t$, $P_A$ with $\kappa_A/\alpha_A$, time with $\alpha_AN_t^2$, defining $\psi=\kappa_A\alpha_B/(\alpha_A^2N_t)$, and $\phi=\kappa_B/(\alpha_A N_t)$ results in
\begin{align}
    \begin{split}
        \dot{G}_A &= P_B G_A\left(1 - G_A-G_B\right)\\
        \dot{G}_B &= \psi P_A G_B\left(1 - G_A-G_B\right)\\
        \dot{P}_A &= G_A\left(1 - G_A-G_B\right)\\
        \dot{P}_B &=\phi G_B\left(1 - G_A-G_B\right).
    \end{split}
    \label{eq:1pool_2sp_cooper}
\end{align}
%
The solution of the system~\ref{eq:1pool_2sp_cooper} is presented in Figure~\ref{fig:1pool_2sp_cooper}.
Here we can see that if $\psi=1$ and $\phi=1$ (see Figure~\ref{fig:1pool_2sp_cooper}~A) the production rate of the public goods of both species is identical to the growth rate of bacteria.
Increasing the production rate of the public good (see Figure~\ref{fig:1pool_2sp_cooper}~B), leads to an increase in the growth of species $A$ that profit from the presence of $B$.
On the other hand, choosing $\psi=2$ (see Figure~\ref{fig:1pool_2sp_cooper}~C), we increase the profit that species $B$ has from public goods $P_A$.
That indirectly leads to larger amount of $P_B$ produced by species $B$.
\begin{figure}
    \begin{center}
    %\includegraphics[width=0.9\columnwidth]{Figures-pool_model_2sp_1pool_coop.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_2sp_1pool_coop1.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_2sp_1pool_coop2.pdf}
    \includegraphics[width=0.9\columnwidth]{Figures-pool_model_2sp_1pool_coop3.pdf}
    \caption{
        The two-species system where each of the species release public goods $P$ is defined with equations~(\ref{eq:1pool_2sp_cooper}) and calculated for different parameters:
        (A) $\psi=1, \phi=1$, (B) $\psi=1, \phi=2$ and (C) $\psi=2, \phi=1$.
        Here the red lines describes the evolution of the species $G_A$ and blue lines correspond to species $G_B$.
        Light brown curve corresponds to public good released by species $A$ ($P_A$) and brown curve shows public goods produced by species $B$ ($P_B$).
    }
    \label{fig:1pool_2sp_cooper}
    \end{center}
\end{figure}

% ================================================================================================
\section{Spatial effects}
As described above, the proposed model operates with the total number of the bacterial counts for
different species and states in the system, not accounting for their spatial distributions.
Assuming that the bacterial system is uniformly distributed (sometimes also called 'well-stirred')
we expect the \ac{ode} description to be valid.
However, in reality spatial effects can play a significant role for population survival forcing some
species to develop competitive mechanisms such as cell adhesion to surfaces in the areas rich with
nutrients~\citep{htuson_bacteriasurface_2013}.
Another mechanism is the motility of bacteria, e.g., the flagellates of \textit{E. coli} or
\textit{B. subtilis} which are capable of chemotactic motion driven by oxygen
concentrations~\citep{decoene_microscopic_2011}.
The bacterial social interactions such as release of toxins or public goods can also
significantly affect the spatial structure of the community~\citep{blanchard_bacterial_2015}.
These effects can only be treated within a spatial framework that is able to describe the observed
effects.\\
Compared to other modeling techniques such as \acp{pde}, \acp{abm}~\citep{nagarajan_agent-based_2022-1} force researchers to think in
terms of cellular processes, thus bridging the gap from a
collective (bulk) model to an individual-based (single-cell) description.
Our goal is to provide an \ac{abm} which can be used to derive the given \ac{ode} model as a
coarse-grained case.
In this way, we can show that the bulk-based \ac{ode} model is directly related to the mechanistic
single-cell behavior of the bacteria.
To this end, we consider the system consisting of two bacterial species that can be described by the two internal states: lag phase and growth.
Both of these species grow using the same resource pool $R$ but with different growth constants
$\alpha_{A/B}$ and different lag phase transition rates $\lambda_{A/B}$.
Moreover, species $A$ is able to release a certain antimicrobial component $I$ that shows an
inhibitory effect on species $B$ (amensalism).
We assume that $L,G$ are measured in terms of the overall volume and $R,I$ are measured as total
amount.
The \acp{ode} system of the bacterial population can be written as:
%
%
% - \citep{emonet_agentcell_2005, gorochowski_bsim_2012, li_nufeb_2019, kreft_bacsim_1998}: examples of another toolboxes for bacterial communities modeling (ABM).\\
% More about the modeling of the motile bacteria can be seen in works:~\citep{decoene_microscopic_2011, rosser_modelling_2014, sokolov_physical_2012, li_amplified_2008}.
%
%
% In general, bacterial communities in 2D can experience collective motion called swarming~\citep{wu_collective_2015}.
% Some works modelled 2D motile bacteria:
% \begin{itemize}
%     \item \citep{decoene_microscopic_2011}: 2D chemotactic self-propelled bacterial motion of the flagellated bacteria (e.g., E. coli, B. subtilis).
%     Chemotaxis here was driven by the oxygen concentrations.
%     \item \citep{rosser_modelling_2014}: modeling of the bacterial reorintation, rotation.
%     \item \citep{sokolov_physical_2012, li_amplified_2008}: modeling of the bacteria using hydrodynamic interactions and Brownian motion.
%     Fluctuations in velocity and rotations.
%     (\citep{li_amplified_2008}: Brownian motion gives right motion observed using microscopy).
%     \item \citep{farrell_mechanically_2013}: 2D modeling using Newtonian dynamics and only mechanical interactions + nutrient diffusion.
%     Branching parameter defines diff limits where diffusion dominant and limits growth, and where it's irrelevant.
%     Mechanical pressure (due to bacteria-bacteria interactions) affects growth, can limit it.(!!!)
% \end{itemize}
%
% For simulation results:
% \begin{itemize}
%     \item \citep{farrell_mechanically_2013}: Showed the influence of mechanic interactions between bacteria in 2D system with bacteria consuming the nutrient.
%     The growth of bacterial colonies where non-motile microorganisms replicate and push each other away as they grow.
%     Found a transition between two different growth regimes, controlled by the balance between growth and uptake of nutrient:\\
%     Exponential growth where diffusion of the nutrient does not play any role. Sublinear growth where growth is limited by nutrient diffusion.
%     (Structurally in 2D it's regimes: smooth propagating of bacterial front vs. branching.)
%     \item \citep{nagarajan_agent-based_2022-1}: review on the ABM for bacterial population.
%     The ABM represent a microbial colony as a group of discrete agents that are governed by a set of rules, where each agent can be an individual cell or a collection of cells.
%     Can use citation when talk about general processes we are accounting as growth, uptake of nutrient, production of inhibitor, e.g.,:
%     Physical interactions between agents are calculated using excluded volume interactions.
%     Chemical interactions as releasing inhibitors are described using concentrations and solved using PDEs.
% \end{itemize}
%
\begin{align}
    \begin{split}
        \dot{L}_A &= -\lambda_A R L_A\\
        \dot{G}_A &= \lambda_A R L_A + \alpha_A R G_A\\
        \dot{L}_B &= -\lambda_B R L_B\\
        \dot{G}_B &= \lambda_B R L_B + \frac{\alpha_B}{1 + \mu_I I} R G_B.
    \end{split}
    \label{eq:spatial_limit_F}
\end{align}
%
Moreover, the microenvironment of our system contains the resource and the inhibitor molecules
\begin{align}
    \begin{split}
        \dot{R} &= -\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{(1 + \mu_I I) N_t} R G_B \\
        \dot{I} &= \kappa G_A.
    \end{split}
    \label{eq:spatial_limit_H}
\end{align}

% ------------------------------------------------------------------------------------------------
\subsection{Constructing an agent-based pool model}
The following sections will demonstrate how a given \ac{ode} system can be translated into a
custom-built \ac{abm} which has only a few more overall parameters.
This procedure is a blueprint which can be followed to construct \acp{abm} for similar
\ac{ode}-based systems.\\
To construct our \ac{abm}, we use the modeling framework \texttt{cellular\_raza}~\citep{Pleyer2025} that allows us to define our bacterial agents from a clean slate.
With this flexibility, researchers are able to construct minimal models using only as many parameters which are necessary.
This ability, to precisely define and control the exact behavior of our cell-agents, enables the following parts.

\subsubsection{Cellular reactions}
To construct an \ac{abm} which resembles the given \ac{ode} system of equations~\eqref{eq:spatial_limit_F} and \eqref{eq:spatial_limit_H}, we consider multiple cellular reactions.
We denote cells of species $A,B$ with $c_{A/B}$ respectively and indicate shared behavior by simply using $c$.
Cells $c_A$ of species $A$ take up extracellular resources $[R_e]$ into the cell $[R_{i,c_A}]$ with rate $u$ in \unit{\per\minute}.
The uptake only occurs at the position of the cell $x_{c_A}$.
Furthermore, we let cells of species $A$ produce a freely diffusible inhibitor $I$ with production rate $b$ and diffusion constant $D_I$ which inhibits cells of species $B$ in their uptake of nutrients.
\begin{equation}
    \dot{[I]} = b\frac{V_{c_A}}{V_D}\delta(x-x_{c_A}) + D_I \Delta [I]
\end{equation}
We model this inhibiting effect by modifying the uptake term of species $B$ with a factor of $\frac{1}{1+\nu_I [I]}$.
% Note that we do not need to distinguish between the two species because we can set parameters individually on agent-level.
% To distinguish between the two species, it suffices to set parameters for every individual agent accordingly.
% This means, we can implement the most general version of the equations and by choosing appropriate parameters, this defines the species of the respective cell-agent.
If we combine these effects and consider multiple cells, we obtain the system
\begin{alignat}{3}
    \dot{[R_{i,c_A}]} &=& u &[R_e] - a[R_{i,c_A}]\\
    \label{eq:spatial-intracellular-a}
    \dot{[R_{i,c_B}]} &=& \frac{u}{1+\nu_I [I]} &[R_e] - a[R_{i,c_B}]\\
    \label{eq:spatial-intracellular-b}
    % TODO fix this
    % \begin{split}
        \dot{[R_e]} &=& u &\sum\limits_{c_A}\frac{V_{c_A}}{V_D} [R_e] \delta(x-x_{c_A})\\
            &&-\frac{u}{1+\nu_I [I]} &\sum\limits_{c_B}\frac{V_{c_B}}{V_D} [R_e] \delta(x-x_{c_B})
    % \end{split}
    \label{eq:spatial-extracellular-resources-pde}
\end{alignat}
where equations~\eqref{eq:spatial-intracellular-a} and \eqref{eq:spatial-intracellular-b} describe the change of intracellular resources for every individual cell of of species $A,B$ and the equation~\eqref{eq:spatial-extracellular-resources-pde} contains extracellular resources coupled to each individual agent.
They contain the uptake of resources and term used grow the cell, converting resources into volume which will be explained below.
We use the Dirac delta-distribution $\delta(x-x_c)$ to enforce exchange of nutrients at the positions of the cells $x_{c}$.
Note that $[R_e]=[R_e](x)$ explicitly depends on the spatial coordinate but we choose to omit its notation.
In equation~\eqref{eq:spatial-extracellular-resources-pde} we need to account for the ratio of cellular $V_c$ and domain volume $V_D$ to calculate the correct exchange.
In addition, we want to modify the cellular volume by growing it increasingly with rate $a$.
Therefore, we need to specify how many resource molecules are turned into volumetric units $\sigma$ in \unit{\micro\metre\cubed\per\mole}.
The total amount of resource molecules present inside the cell is given by the current concentration multiplied by the cell-volume $V_c [R_{i,c}]$.
With a rate $a$ and conversion $\sigma$, the cell turns these resource molecules into volume.
\begin{equation}
    \dot{V}_c = a\sigma [R_{i,c}] V_c
\end{equation}
In order to simplify this system, we can assume that the internal resources are constantly being depleted by the growth and the limiting process is the uptake of the resource.
This means, we can effectively set $\dot{[R_{i,c}]}=0$.
Therefore, we obtain the relations $\frac{u}{1+\nu_I [I]}[R_e] = a[R_{i,cB}]$ and $u[R_e]=a[R_{i,c_A}]$ which we can use to eliminate the variable monitoring internal resource concentrations $[R_{i,c}]$.
Additionally, we also require the nutrients to be freely diffusible with constants $D_R$.
With these considerations, we can update equations~\eqref{eq:spatial-intracellular-a} to \eqref{eq:spatial-extracellular-resources-pde} and obtain:
\begin{align}
    \begin{split}
        % TODO format these equations
        \dot{V}_{c_B} &= u \sigma [R]V_c\\
        \dot{V}_{c_A} &= \frac{u}{1+\nu_I [I]} \sigma [R]V_c\\
        % \dot{[R_e]} &= \sum\limits_c -\frac{u}{1+\nu_I [I]} \frac{V_c}{V_D} [R] \delta(x-x_c) + D_R\Delta [R]\\
        \dot{[R]} &= u \sum\limits_{c_A}\frac{V_{c_A}}{V_D} [R] \delta(x-x_{c_A})\\
            &-\frac{u}{1+\nu_I [I]} \sum\limits_{c_B}\frac{V_{c_B}}{V_D} [R] \delta(x-x_{c_B})\\
            &+D_R \Delta [R]\\
        \dot{[I]} &= \sum\limits_c b\frac{V_c}{V_D}\delta(x-x_c) + D_I\Delta [I]
    \end{split}
    \label{eq:spatial-full-system-simplified}
\end{align}
For the diffusion, we require Neumann (zero-flux) boundary conditions.
Since, we will not explicitly model intracellular concentrations $[R_{i,c}]$, we will drop the index entirely, only considering the extracellular resource.
%
%
% ................................................................................................
\subsubsection{Physical mechanics \& interactions}
We model cells as spherical objects in 2D space.
They can be represented by their position $x_c$ and radius $r_c$ which is related to their (2D) volume $V_c = \pi r_c^2$.
As the coupled reactions from equations~\eqref{eq:spatial-full-system-simplified} change $V_c$, we also automatically adapt the cell radius accordingly.
% Notice however, that how the radius is updated does not correspond to the cellular reactions defined previously but is a purely mechanical consideration.
The dimensionality of the problem does not find its way into the cellular reactions but only manifests itself in these mechanical considerations.
The dynamics are described by Newton's law of motion for particle damping.
\begin{equation}
    m \ddot{\mathbf{x}} = -\eta\dot{\mathbf{x}} + \mathbf{F}.
\end{equation}
Cells are interacting via a point-like interaction potential.
We can calculate the interaction strength via
\begin{alignat}{3}
    \mathbf{z} &= \mathbf{x_1} - \mathbf{x_2} \hspace{2em}&& r = ||\mathbf{z}||\\
    \sigma &= \frac{r}{r_1+r_2} && q=0.2\\
    \mathbf{F} &= \epsilon \frac{1+q}{q+\sigma} \theta(\sigma - 1) \frac{1}{r}\mathbf{z}
\end{alignat}
We assume that the cells are not sticky and consider only a repulsive interaction potential, leaving out any possible attractive forces~\citep{Garrett2008}.
These mechanisms will result in the overall expansion of the colony as individual cells push each other further apart.
%
% \begin{figure}
%     \centering
%     \includegraphics[width=0.6\textwidth]{Figures-interaciton_potential.png}
%     \caption{
%         Overall form of the interaction force between two cells.
%         A value $\sigma<1$ indicates that cells are overlapping which results in a positive force outwards.
%         Conversely, as long as cells are apart $\sigma>=1$, no force is acting.
%     }
%     \label{fig:spatial-interaction-potential}
% \end{figure}
%
%
% ................................................................................................
\subsubsection{Cell cycle}
Typically the cycle of a cell consist of multiple phases~\citep{Cooper2000_fn}.
Since our goal is to include only the necessary details of the underlying biological problem, we drastically simplify the description of the cell cycle.
Cell growth is already being described by the system of coupled \acp{ode} given in equations~\eqref{eq:spatial-full-system-simplified}.
Thus, we introduce an additional rule that performs the division process when a certain volume threshold is reached.
The volume of the cell is divided equally between the new and old cell and both are assigned new random positions close to each other such that their radii do touch but without any overlap.
\begin{figure}
    \begin{center}
    \begin{tikzonimage}[width=0.49\columnwidth]
        {Figures-snapshot-2-cells-before-division.png}%
        \node at (0.01, 0.99)[anchor=north west]{\textbf{A}};
    \end{tikzonimage}%
    \hspace{0.01\columnwidth}%
    \begin{tikzonimage}[width=0.49\columnwidth]
        {Figures-snapshot-2-cells-after-division.png}
        \node at (0.01, 0.99)[anchor=north west]{\textbf{B}};
    \end{tikzonimage}
    \caption{
        (A) A cell right before and (B) after a division event.
        The volume is equally distributed between the two cells and the new positions are picked
        random but at opposing sides on a circle around the original center of the single cell.
    }
    \end{center}
\end{figure}
In addition to modeling division events, the cells are subject to the lag-phase which delays their ability to proliferate.
We model this by assigning a transition rate to each cell.
For each time point of the simulation, we generate a boolean value from a random number generator with probability
\begin{equation}
    P(\text{transition}) = \Delta t \lambda \frac{[R]}{[R_0]}.
\end{equation}
If this boolean is true, we move the cell from the inactive state to the active state where it starts to consume nutrients and grow.
The concatenation of many such stochastic processes on a population-based level converges identically to the lag-phase behavior of the \ac{ode} model.
%
%
% ------------------------------------------------------------------------------------------------
\subsection{Comparing \acs{abm} and \acs{ode} models}
The \ac{abm} model should be considered to be more fundamental compared to the \ac{ode} description in the sense that it incorporates more biological aspects and maps them closer to the actual behavior of the bacteria.
We are thus able to derive the \ac{ode} pool model from its \ac{abm} version via a process of homogenization (see Appendix~\ref{sec:supplement-abm-derivation}).
By doing so, we find a mapping between parameters from the \ac{abm} to parameters used in the \ac{ode} description.
The derived equations read
\begin{align}
<<<<<<< HEAD
    \dot{L}_A &= -\lambda L_A\frac{R}{R_0}\\
    \dot{L}_B &= -\lambda L_B\frac{R}{R_0}\\
    \dot{V}_A &= u\sigma R \frac{V_A}{V_D} + \lambda L_A\frac{R}{R_0}\\
    \dot{V}_B &= \frac{u}{1+\nu_I V_D^{-1} I} \sigma R \frac{V_B}{V_D} + \lambda L_B\frac{R}{R_0}\\
    \dot{R} &= - R \frac{V_A}{V_D} - \frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V_B\\
=======
    \dot{L_A} &= -\lambda L_A\frac{R}{R_0}\\
    \dot{L_B} &= -\lambda L_B\frac{R}{R_0}\\
    \dot{V_A} &= u\sigma R \frac{V_A}{V_D} + \lambda L_A\frac{R}{R_0}\\
    \dot{V_B} &= \frac{u}{1+\nu_I V_D^{-1} I} \sigma R \frac{V_B}{V_D} + \lambda L_B\frac{R}{R_0}\\
    \dot{R} &= - u R \frac{V_A}{V_D} - \frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V_B\\
>>>>>>> e83cdaa63bd845a72cd40759475fffbd002cf9aa
    \dot{I} &= b V_A
\end{align}
and the \ac{ode} equations are
\begin{align}
    \dot{L}_A &= -\lambda_A R L_A\\
    \dot{L}_B &= -\lambda_B R L_B\\
    \dot{G}_A &= \lambda_A R L_A + \alpha_A R G_A\\
    \dot{G}_B &= \lambda_B R L_B + \frac{\alpha_B}{1 + \mu_I I} R G_B\\
    \dot{R} &= -\frac{\alpha_A}{N_t} R G_A-\frac{\alpha_B}{(1 + \mu_I I) N_t} R G_B \\
    \dot{I} &= \kappa G_A
\end{align}
When comparing these equations, we can map them onto each other via the relations
\begin{align}
    \begin{split}
        N_t &= [R_0]V_D\sigma\\
        \mu_I &= \frac{\nu_I}{V_D}\\
        \alpha &= u[R_0]\sigma\\
        \kappa &= \sigma.
    \end{split}
    \label{eq:ode-abm-mappings}
\end{align}
Furthermore, we can use these mappings to numerically compare both approaches.
Table~\ref{tab:spatial_limit_init} shows an example comparison of parameters of the \ac{ode} and \ac{abm} which have been mapped onto each other.
Figure~\ref{fig:spatial-snapshots} shows snapshots of a numerical simulation of the described system.
The initial bacteria have been ordered on a grid such that only every diagonal neighbor is of the same species, thus resulting in a more homogenous initial distribution.
%
\begin{figure}
    \begin{tikzonimage}[width=0.49\columnwidth]
        {Figures-abm-homogenous-images-snapshot_00006000.png}
        \node at (0.01, 0.99)[anchor=north west]{\textbf{A}};
    \end{tikzonimage}%
    \hspace{0.01\columnwidth}%
    \begin{tikzonimage}[width=0.49\columnwidth]
        {Figures-abm-homogenous-images-snapshot_00012000.png}
        \node at (0.01, 0.99)[anchor=north west]{\textbf{B}};
    \end{tikzonimage}\\
    \begin{tikzonimage}[width=0.49\columnwidth]
        {Figures-abm-homogenous-images-snapshot_00018000.png}
        \node at (0.01, 0.99)[anchor=north west, white]{\textbf{C}};
    \end{tikzonimage}%
    \hspace{0.01\columnwidth}%
    \begin{tikzonimage}[width=0.49\columnwidth]
        {Figures-abm-homogenous-images-snapshot_00024000.png}
        \node at (0.01, 0.99)[anchor=north west, white]{\textbf{D}};
    \end{tikzonimage}%
    \caption{
        Simulation snapshots at $1.04$~\unit{\day}, $2.08$~\unit{\day}, $3.13$~\unit{\day} and $4.17$~\unit{\day} corresponding to the parameters given in Table~\ref{tab:spatial_limit_init}.
        Species $B$ is colored blue while species $A$ is colored in red.
        The background shows the resource concentration $[R]$.
        Darker colors indicate that less nutrients are present.
    }
    \label{fig:spatial-snapshots}
\end{figure}
%
%
Figure~\ref{fig:spatial-growth-curve} shows multiple results for various configurations of the \ac{abm}, differing only in their initial spatial configuration and diffusivity.
We see that in the highly-diffusive homogenous case of subfigure (A), the pool model is a good fit to the \ac{abm} model.
Heterogeneity and slower diffusion cause larger differences between the two models as can be seen in subfigures (B) and (C).
The combination of the two effects shows the largest difference between model outputs.
In subfigure (C), due to the heterogeneous cell distribution (see Appendix~\ref{fig:supplement-spatial-initial-config-cells}), overall inhibition of species $B$ is significantly reduced.
Subfigure (D) shows how a randomized homogenous initial distribution can lead to even more notable effects of inhibition.
In contrast to the other results, here the final values of species $A$ and $B$ have larger/smaller counts respectively.
This illustrates that spatial effects can shift the final results of growth curves towards both directions, leading to an overall increase/decrease of one species.
We furthermore note that the total bacteria volume of the \ac{abm} model is always smaller or equal compared to the \ac{ode} model.
This is expected, since processes such as diffusion and cells growing outward introduce another time component which is simply assumed to be instantaneous in the \ac{ode} model.
The chosen diffusion constants $D_I=D_R=30.0$~\unit{\micro\metre\squared\per\minute} are in the order of small molecules such as $NaCl$ while the lower values $D_I=D_R=5.0$~\unit{\micro\metre\squared\per\minute} are around values of Glucose in water.
\begin{figure*}
    \centering
    \begin{tikzonimage}[width=\columnwidth]
        {Figures-abm-homogenous-abm_ode_comparison.png}
        \node at (0, 1)[anchor=north west]{\textbf{A}};
    \end{tikzonimage}%
    \begin{tikzonimage}[width=\columnwidth]
        {Figures-abm-homogenous-low-diffusion-abm_ode_comparison.png}
        \node at (0, 1)[anchor=north west]{\textbf{B}};
    \end{tikzonimage}\\
    \begin{tikzonimage}[width=\columnwidth]
        {Figures-abm-inhomogenous-abm_ode_comparison.png}
        \node at (0, 1)[anchor=north west]{\textbf{C}};
    \end{tikzonimage}%
    \begin{tikzonimage}[width=\columnwidth]
        {Figures-abm-random-abm_ode_comparison.png}
        \node at (0, 1)[anchor=north west]{\textbf{D}};
    \end{tikzonimage}
    \caption{
        (A-D) Comparison of the \ac{ode} and \ac{abm} model for different configurations, showing homogenous, heterogeneous or randomized initial configurations.
        The diffusion constants are chosen such that $D_R=D_I=D$.
        Snapshots of the spatial configurations of cells are included in the Appendix~\ref{fig:supplement-spatial-initial-config-cells}.
    }
    \label{fig:spatial-growth-curve}
\end{figure*}
These results show how variations in properties that can not be described by simple \ac{ode} models can lead to large deviations in the growth curves of the species.
Furthermore, it was shown that the utilized pool model can be derived from first principles by performing a homogenization of the mechanistic bacterial properties.
%
%
% ================================================================================================
\section{Discussion}

% How well does the pool model do to encapsulate all the different growth models
% \begin{itemize}
%     \item All 6 models can be represented by pool model
%     \item Baranyi-Roberts results can be reproduced, but not identical; requires some work
%     \item what are some general limitations within this framework? how can they be circumvented?
%     \item why is the pool model "better" than the other \ac{ode} models? (why did we do this?);
%         generalization of many models; better explainability;
%     \item link to supplement; we can generalize this and prove theorems for the abstract,
%         generalized model; this allows us to discuss many cases at once
% \end{itemize}
%
% How can we cover more effects of bacteria?
% \begin{itemize}
%     \item More states for metabolic processes
%     \item More states for more species
%     \item Other cooperiation/inhibition effects
%     \item external influences
% \end{itemize}
%
% How justified was the construction of the \ac{abm}?
% \begin{itemize}
%     \item Discuss simplified physical interaction model
%     \item Heterogeneity plays a role; as we have shown
%     \item why did we not simply use a PDE?; because we have to think in terms of cellular processes
%         and this ensures that our model is actually realistic
% \end{itemize}
%
% What are limitations of our approach?
% \begin{itemize}
%     \item spatial stuff; heterogeneity (because \ac{ode})
%     \item larger models require more parameters, harder to estimate
%     \item only simple models have analytical solutions
% \end{itemize}

The presented pool model was used to describe the dynamics of one- and two-species systems containing the main growth stages and interactions between bacteria in the form of cooperation and competition.
Our approach produces results which exhibit similarities to other well-known predictive microbiology models such as the Baranyi-Roberts model while providing better explainability.
The inclusion of additional metabolic processes or larger number of cell-species can be achieved by extending the list of possible states and defining transitions between them.
As an example, the description of the quorum sensing phenomena can be modelled by adding the equation for the released autoinducer molecules as a part of the microenvironment.
These molecules then affect the same-species bacteria growth controlling the size of the population.
The approach described above provides a primary predictive model with transition rates between different pools as parameters.
To include the influence of external factors, i.e., temperature, pH, pressure, etc., one can simply use the secondary models for the transition rates, e.g., the square-root temperature dependence~\citep{ratkowsky_relationship_1982}.\\
By constructing an \ac{abm}, we have exemplified how close the devised model is to the individual biological 
processes that occur within the bacteria.
This mechanistic single-cell level of understanding of our \ac{abm} is a perspective which can not be easily replicated by other spatial models such as \acp{pde}.
We have investigated the impact of spatial effects in the form of varying diffusion parameters and different initial spatial distributions of bacteria.
Our findings reinforce that the \ac{ode} model is a good simplification in the highly-diffusive regime within a mostly homogenous spatial bacterial distribution.
However, deviations to these assumptions can lead to non-negligible effects, influencing the growth of the colony.

% ------------------------------------------------------------------------------------------------
\section{Conclusion}
% \begin{itemize}
%     \item New theoretical framework
%     \item consistent with \ac{ode} models (Baranyi-Roberts)
%     \item Better explainability
%     \item Include effects such as inhibition, toxins in understandable manner
%     \item showed spatial limitations of \ac{ode} model
%     \item \ac{abm} model allowed modeling of cellular processes; ensure that pool model is derivable
%         from the individual behvaiour of cellular processes
%     \item outlook: generalized mathematical formulation; extend inhibition model (because it is a
%         dimensional reduction of existing models)
%     \item Provide code in a pypi package as well (TODO)
% \end{itemize}

We introduced the pool model as a framework to unify existing approaches and allow researchers to follow a mechanistic approach during model development.
It is able to capture the effects of many established models and provides a lens which is useful to generalize existing systems and results in better overall explainability.
By modeling effects such as mutual inhibition on a single-cell level within an \ac{abm}, we have exemplified that the pool model directly builds on these more fundamental concepts.
It can thus aid in the mechanistic understanding of the underlying processes, allowing researchers to deepen their knowledge about microbial growth within food science.
Going forward, this framework can provide a starting point for a more abstract mathematical formulation which should be able to cover a much wider class of systems.

\onecolumn

\newpage
\section*{Data availability \& reproducibility}

The data and simulation code which are presented in this study are publicly available in
\url{https://github.com/polinagaindrik/pool_model_paper}.

% \printbibliography
\bibliography{references.bib}

% ================================================================================================
\newpage
\beginsupplement
% ================================================================================================
\section{General mathematical formalism}
\label{ssec:supplement1}
Consider the system consisting of $N$ species in $M$ different states.
In the general representation, all discussed in the main part systems  can be described with the Ordinary Differential Equations (\acp{ode}) using a non-linear function $\mbF$:
\begin{equation}
   \dot{\mbx} = \mbF(\mbx, \mbm, \mbp, \mbu),
\label{eq:model_ODE_x}
\end{equation}
where  $\mbx \in \mathbb{R}^{N}  \otimes \mathbb{R}^{M}$ is a vector describing bacterial count.
In matrix form it can be written as
\begin{equation}
    \mbx = \begin{pmatrix}
        x^1_1  & \dots & x^1_N  \\
        \vdots &       & \vdots \\
        x^M_1  & \dots & x^M_N  \\
            \end{pmatrix}
    \label{eq:model_bact}
\end{equation}
with $x_{j}^{i}$ describing the bacterial counts for species $i$ in state $j$.
$\mbu$ is the vector of different inputs of the model, e.g., temperature, pressure etc.
The vector $\mbm \in \mathbb{R}^{K}$ includes all variables corresponding to the microenvironment of the system which can be concentration of the inhibitors or activators:
\begin{equation}
    \dot{\mbm} = \mbH (\mbx, \mbm, \mbp, \mbu).
    \label{eq:model_microenv}
\end{equation}
Here function $\mbH$ is in general case also non-linear.\\
%
The advantage of the following formulation is its generality.
Depending on the knowledge about the system as well as the needed level of accuracy one can define an arbitrary number of states $M$ and control the complexity of the model.
Moreover, opposite to the already known Predictive Microbiology like Baranyi and Roberts one can easily change.\\
%
To calculate the bacterial count for one species $i$, we simply need to sum the tensor/vector $\mbx$ over all possible states:
\begin{equation}
    N_i = \sum_k \mbx_i^k.
\label{eq:N_pro_species}
\end{equation}
While the total bacterial count of all species of our system is
\begin{equation}
    N = \sum_i \sum_k \mbx_i^k.
\label{eq:N_total}
\end{equation}
%
%
%The three pool system described with equations~\ref{eq:ode_3pools} will be written as
%\begin{equation}
%    \dot \mbx  = A \mbx, \hspace{3mm} A = \begin{pmatrix}
%        -(\mu + \lambda) & 0             & 0      \\
%        \lambda          & \alpha - \mu' & 0      \\
%        \mu              & \mu'          & -\beta 
%    \end{pmatrix}.
%\label{eq:x_repres_3pools}
%\end{equation}
%
Using such matrix representation of the bacterial system, we can rewrite, for example, the three-pool model with limiting resource described with equations~\ref{eq:ode_3pool_resource} in a more compact form.
The concentrations of the pools are then the components of our variable vector $\mbx (t) = (L(t), G(t), D(t))^T$.
%
\begin{align}
    \dot \mbx  &= \mbF(\mbx, \mbm, \mbp, \mbu) = \begin{pmatrix}
        -(\mu + \lambda) & 0               & 0      \\
        \lambda          & \alpha R - \mu' & 0      \\
        \mu              & \mu'            & -\beta 
    \end{pmatrix} \mbx\\
    \dot \mbm &= \mbH (\mbx, \mbm, \mbp, \mbu) = -\frac{\alpha}{N_t} R G.
\end{align}
Separating linear and non-linear parts in two terms yields
\begin{equation}
    \dot \mbx = \begin{pmatrix}
        -(\mu + \lambda) & 0       & 0 \\
         \lambda         & \alpha  & 0 \\
         \mu &  \mu'& - \beta 
    \end{pmatrix} 
    \mbx + \frac{\alpha}{N_t} x^2 \sum_i^3 x^i \begin{pmatrix} 0 \\ 1 \\ 0  \end{pmatrix}.
\label{eq:x_repres_3pools_resource2}
\end{equation}
Here the first term is the linear part and the second term define the quadratic behavior of the system.
%
\paragraph{Generalized Lotka-Volterra}

Likewise, the multispecies system with resource competition present described by equations~\ref{eq:model_2sp_resource_comp} can be represented in the same way by defining
\begin{equation}
    \mbm = R, \hspace{1cm}
    \mbx = \begin{pmatrix}
        L_A & L_B \\
        G_A & G_B 
    \end{pmatrix}.
\label{eq:gLV_x_m}
\end{equation}

The simpler version of one state per species (equation~\ref{eq:LV_simple}) can be simply written in a general mathematical form as generalized Lotka-Volterra:
\begin{equation}
    \frac{d x_i}{d t} = x_i f_i (\mbx), \hspace{3mm} \mathbf{f} = \mathbf{r} + A \mbx.
\label{eq:gLV}
\end{equation}
%
%
\vfill
\pagebreak
\section{Deriving the pool model}
\label{sec:supplement-abm-derivation}
\paragraph{Reactions}
Since the \ac{abm} can be considered more fundamental than the \ac{ode} model, we should be able to derive the latter one from the former.
This can be achieved by starting with equations~\eqref{eq:spatial-full-system-simplified}.
We integrate the spatial equations over the total domain starting with the resource concentration $[R]$.
\begin{alignat}{8}
    \int\limits_D\dot{[R]}\text{dVol}
    &= \int\limits_D\bigg[&-&u \sum\limits_{c_A}\frac{V_{c_A}}{V_D} [R] \delta(x-x_{c_A})
        &-&\frac{u}{1+\nu_I [I]} \sum\limits_{c_B}\frac{V_{c_B}}{V_D} [R] \delta(x-x_{c_B})
        +D_R \Delta [R]\bigg]\text{dVol}\\
    &= &-&u\sum\limits_{c_A}V_{c_A}\int\limits_D [R] \frac{\delta(x-x_{c_A})}{V_D} \text{dVol}
        &-&\frac{u}{1+\nu_I [I]} \sum\limits_{c_B}V_{c_B} \int\limits_D [R] \frac{\delta(x-x_{c_B})}{V_D}\text{dVol}
        +D_R \int\limits_{\partial D} \nabla[R]\text{dVol}\\
    &= &-&u\sum\limits_{c_A} V_{c_A} [R]_{x=x_{c_A}}
        &-& \frac{u}{1+\nu_I [I]} \sum\limits_{c_B} V_{c_B} [R]_{x=x_{c_B}}\\
    &= &-&u\sum\limits_{c_A} V_{c_A} [R]
        &-& \frac{u}{1+\nu_I [I]} \sum\limits_{c_B} V_{c_B} [R]\\
    &= &-&uV_A [R] &-&\frac{u}{1+\nu_I[I]} V_B [R]\\
    &= &-&u \frac{V_A}{V_D}R &-&\frac{u}{1+\nu_I V_D^{-1} I} \frac{V_B}{V_D}R
\end{alignat}
We can split up the integral into three parts.
The last part can then be transformed into an integral over the flux of the resource along the boundary.
Since we chose Neumann boundary conditions, this contribution vanishes identically.
Afterwards, the remaining parts can be evaluated by using the delta-distribution.
Note that we additionally need to consume a factor of $V_D^{-1}$ which is the inverse of the domain
volume when evaluating the delta function.
We evaluate the resource concentration $[R]$ at the position of the individual cells and sum over all contributions.
Now we assume homogeneity in the sense, that we can reduce $[R]$ to only a variable with time-dependence, with homogeneous spatial distribution.
This means that every contribution does not depend on the position at which it was evaluated, i.e., $[R]_{x=x_c}=[R]$.
Finally, we can perform the sum to gain the total volume of all cells $V_{A/B}=\sum\limits_c V_{c_{A/B}}$.
Following the same principle as before, we can integrate the inhibitor equation.
\begin{alignat}{7}
    \int\limits_D\dot{[I]} &= \int\limits_D \bigg[&&\sum\limits_{c_A} b\frac{V_{c_A}}{V_D}&&\delta(x-x_{c_A}) &&+ D_I\Delta [I]\bigg]\text{dVol}\\
    &= \int\limits_D&&\sum\limits_{c_A} b\frac{V_{c_A}}{V_D}&&\delta(x-x_{c_A}) \text{dVol} &&+ \int\limits_D D_I\Delta [I]\text{dVol}\\
    &= \int\limits_D&&\sum\limits_{c_A} b\frac{V_{c_A}}{V_D}&&\delta(x-x_{c_A}) \text{dVol} &&+ \int\limits_{\partial D} D_I\nabla [I]\mathbf{\text{dA}}\\
    &= &&\sum\limits_{c_A} b V_{c_A}\int\limits_D \frac{1}{V_D} &&\delta(x-x_{c_A}) \text{dVol}
\end{alignat}
%
\paragraph{Lag-Phase}
Cells are initially assumed to be in a lag-phase which they leave with probability $P(\text{transition})$.
Dividing this probability by the time increment $\Delta t$, gives us the rate at which the transition is occuring.
This expression can be transfered to a collective rate of change by multiplying with the volume of cells in lag-phase $L_{A/B}$.
\begin{align}
    P(\text{transition}) &= \Delta t \lambda \frac{[R]}{[R_0]}\\
    \frac{1}{\Delta t}P(\text{transition}) &= \lambda \frac{[R]}{[R_0]}\\
    \frac{L_{A/B}}{\Delta t}P(\text{transition}) &= \lambda L_{A/B}\frac{R}{R_0}\\
    \partial_t L_{A/B} &= -\lambda L_{A/B}\frac{R}{R_0}
\end{align}
In the last step, we take the expectation value and equate it with the negative of its derivative $\partial_t L_{A/B}$.
%
%
\paragraph{Growth}
For the equation describing the growth of the cell volume, we simply add up cells for the respective species to obtain their total volume.
\begin{alignat}{5}
    \sum\limits_{c_A}\dot{V}_{c_A}
    &= &\sum\limits_{c_A} u& \sigma [R]V_{c_A} + \lambda L_A\frac{R}{R_0} &= &&\sigma [R]V_A + \lambda L_A\frac{R}{R_0}\\
    \sum\limits_{c_B}\dot{V}_{c_B}
    &= &\frac{u}{1+\nu_I [I]} &\sigma [R]V_B + \lambda L_B\frac{R}{R_0} &= &\sum\limits_{c_B} \frac{u}{1+\nu_I [I]} &\sigma [R]V_{c_B} + \lambda L_B\frac{R}{R_0}
\end{alignat}
In addition, we can reformulate $\tilde{\sigma}=\sigma/V_D$ thus redefining the conversion rate $\sigma$ in terms of the total volume $V_D$.
We can thus summarize the equations obtained
\begin{align}
    \dot{L}_A &= -\lambda L_A\frac{R}{R_0}\\
    \dot{L}_B &= -\lambda L_B\frac{R}{R_0}\\
    \dot{V}_A &= u\sigma R \frac{V_A}{V_D} + \lambda L_A\frac{R}{R_0}\\
    \dot{V}_B &= \frac{u}{1+\nu_I V_D^{-1} I} \sigma R \frac{V_B}{V_D} + \lambda L_B\frac{R}{R_0}\\
    \dot{R} &= - R \frac{V_A}{V_D} - \frac{u}{1+\nu_I V_D^{-1}I} \frac{1}{V_D} R V_B\\
    \dot{I} &= b V_A
\end{align}
In this way, we are able to recover the \ac{ode} model.
%
%
% ................................................................................................
\vfill
\pagebreak
\section{Full list of parameters}
We utilize the calculate mappings (see equations~\eqref{eq:ode-abm-mappings}) between the \ac{ode} formulation and \ac{abm} to compare values for parameters which serve a similar purpose within the two simulations.
Table~\ref{tab:spatial_limit_init} lists these parameters together with their values.
Most parameters have clear $1:1$ mappings to ones within the other model.
However, some such as the ones related to spatial dynamics (damping $\eta$, mass $m$, potential
strength $\epsilon$, diffusion $D_R,D_I$) do not.
\begin{table}[H]
    \centering
    \begin{tabularx}{\textwidth}{@{}l *2{>{\centering\arraybackslash}X} l *2{>{\centering\arraybackslash}X}@{}}
        \multicolumn{3}{c}{\textbf{\acs{ode}}} & \multicolumn{3}{c}{\textbf{\acs{abm}}}\\
        \toprule
        \multicolumn{6}{c}{\textbf{Cellular Reactions}}\\
        \toprule
        Production $G_A$            & $\alpha_A$    & $0.002$~\unit{\per\minute}
        & Uptake                    & $u_A$         & $0.01$~\unit{\per\minute}\\
        Production $G_B$            & $\alpha_B$    & $0.002$~\unit{\per\minute}
        & Uptake                    & $u_B$         & $0.01$~\unit{\per\minute}\\
        Production $I$              & $\kappa$      & $0.1$~\unit{\per\minute}
        & Production $I$            & $b$           & $0.1$~\unit{\per\minute}\\
        Inhibition                  & $\mu_I$       & $1.0\cdot 10^{-7}$~\unit{\per\micro\metre\cubed}
        &                           & $\nu_I$       & $0.1$~\unit{\micro\metre\cubed\per\mol}\\
        &&
        & Conversion                & $\sigma$      & $0.1$~\unit{\micro\metre\cubed\per\mol}\\
        Total volume                & $N_t$& $2.0\cdot 10^5$~\unit{\micro\metre\cubed}
        &&&\\
        &&
        & Diffusion $R$             & $D_R$         & $30.0$~\unit{\micro\metre\squared\per\minute}\\
        &&
        & Diffusion $I$             & $D_I$         & $30.0$~\unit{\micro\metre\squared\per\minute}\\
        % Note: this diffusion roughly corresponds to the one of ethanol in water
        % (see https://de.wikipedia.org/wiki/Diffusionskoeffizient#In_Fl%C3%BCssigkeiten)
        % 15.0 m^2/min = 0.9 m^2/s
        % Ethanol: 0.84 m^2/s
        \toprule
        \multicolumn{6}{c}{\textbf{Physical Mechanics \& Interactions}}\\
        \toprule
        &&& Damping                 & $\eta$        & $0.5$~\unit{\per\minute}\\
        &&& Mass                    & $m$           & $7.70$~\unit{\pico\gram}\\
        &&& Potential               & $\epsilon$    & $0.5$~\unit{\pico\gram\micro\metre\per\minute\squared}\\
        \toprule
        \multicolumn{6}{c}{\textbf{Cell Cycle}}\\
        \toprule
        Rate $L_A\rightarrow G_A$   & $\lambda_A$   & $0.0025$~\unit{\per\minute}
        &                           & $\lambda_A$   & $0.0025$~\unit{\per\minute}\\
        Rate $L_B\rightarrow G_B$   & $\lambda_B$   & $0.0005$~\unit{\per\minute}
        &                           & $\lambda_B$   & $0.0005$~\unit{\per\minute}\\
        &&& Division Vol.           & $V_t$         & $14.14$~\unit{\micro\metre\cubed}\\
        \toprule
        \multicolumn{6}{c}{\textbf{Initial Values}}\\
        \toprule
        Volume of $L_A$             & $L_A$         & $127.2$~\unit{\micro\metre\cubed}
        & Cell Number               & $C_A$         & 18\\
        Volume of $G_A$             & $G_A$         & $0.0$~\unit{\micro\metre\cubed}
        & Cell Volume               & $V_{c,A}$     & $7.07$~\unit{\micro\metre\cubed}\\
        Volume of $L_B$             & $L_B$         & $127.23$~\unit{\micro\metre\cubed}
        & Cell Number               & $C_B$         & 18\\
        Volume of $G_B$             & $G_B$         & $0.0$~\unit{\micro\metre\cubed}
        & Cell Volume               & $V_{c,B}$     & $7.07$~\unit{\micro\metre\cubed}\\
        Relative Nutrients          & $R$         & $1.0$
        & Nutrients                 & $[R_0]$       & $1.0$~\unit{\mol\per\micro\metre\cubed}\\
        Inhibitor                   & $I_0$         & $0.0$~\unit{\mol}
        & Inhibitor                 & $[I_0]$       & $0.0$~\unit{\mol\per\micro\metre\cubed}\\
        \bottomrule
    \end{tabularx}
    \caption{
        Comparison of the parameters for the \ac{ode} model and \ac{abm}.
        % To view, how these parameters are stored inside a cell-agent, we refer to the supplementary
        % material~\ref{sec:supplement-full-parameters}.
    }
    \label{tab:spatial_limit_init}
\end{table}
%
% ................................................................................................
\vfill
\pagebreak
\section{Homogeneous and inhomogeneous initial cell distributions}
To clarify what we mean by homogeneous and inhomogeneous initial configurations, we include snapshots of the simulation.
At this point, cells have already grown such that the sizes of cell-clumps are better visible by the naked eye.
The snapshots were taken after $2.08$~\unit{\day}.
\begin{figure}[H]
    \centering
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures-abm-homogenous-images-snapshot_00012000.png}
        \subcaption{Homogeneous initial cell distribution, $D_R=D_I=30.0$~\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}%
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures-abm-homogenous-low-diffusion-images-snapshot_00012000.png}
        \subcaption{Homogeneous initial cell distribution, $D_R=D_I=5.0$~\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}\\
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures-abm-inhomogenous-images-snapshot_00012000.png}%
        \subcaption{Heterogeneous initial cell distribution, $D_R=D_I=30.0$~\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}%
    \begin{subfigure}[c]{0.5\textwidth}
        \includegraphics[width=\textwidth]{Figures-abm-random-images-snapshot_00012000.png}%
        \subcaption{Random initial cell distribution, $D_R=D_I=30.0$~\unit{\micro\metre\squared\per\minute}}
    \end{subfigure}
    \caption{
        Spatial configuration of cells corresponding to the pictures shown in figure~\ref{fig:spatial-growth-curve}.
    }
    \label{fig:supplement-spatial-initial-config-cells}
\end{figure}
%
%
\end{document}
